---
output: pdf_document
params:
  ID: ""
  script_dir: ""
  exome_length: ""
  type: ""
  primary_tm: ""
  tumor_sample: ""
header-includes:
- \usepackage{titling}
- \pretitle{\begin{center}\LARGE\includegraphics[width=160pt]{`r file.path(params$script_dir, "NOT_logo.png")`}\\[\bigskipamount]}
- \posttitle{\end{center}}
- \usepackage{pdflscape}
- \newcommand{\blandscape}{\begin{landscape}}
- \newcommand{\elandscape}{\end{landscape}}
- \usepackage{float}
- \usepackage{booktabs}
- \usepackage[table]{xcolor}
title: \ \ NeuroOncology Technologies (NOT)
author: "Whole Exome Sequencing Report"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
options(stringsAsFactors = FALSE, Encoding="UTF-8")
library(knitr)
library(kableExtra)
```

\pagenumbering{gobble} 
\vspace{2cm}

***

>>> `r paste("**NOT ID: **", params$ID)` \
**Treatment status:** `r ifelse(as.logical(params$primary_tm), "Primary (untreated) tumor", "Recurrent (previously treated) tumor")` \
**Tumor sample:** `r ifelse(params$tumor_sample == "FFPE", "Formalin-Fixed Paraffin-Embedded (FFPE) tissue specimen", "Fresh tumor tissue, snap-frozen in liquid nitrogen")` \
**Normal sample:** Peripheral venous blood \
**DNA extraction method:** QIAGEN DNeasy Blood & Tissue kit \
**Report date:** `r format(Sys.time(), '%d %B, %Y')`

***


\raggedright
\clearpage
\pagenumbering{arabic}


# I. Quality Metrics

## Ia. Summary Table of Quality Metrics

```{r QC, echo = F}
QC_df <- read.csv("./QC_table.csv" )
colnames(QC_df)[1] <- ""
kable_styling(kable(QC_df,
                    format = "latex", 
                    booktabs = TRUE,
                    align = c("r", "c", "c")), 
              latex_options = "striped",
              stripe_color = "lightgray")
```

>>> \* PF is defined as passing Illumina's filter \
    \*\* The percentage of PF reads that aligned to the reference sequence \
    \*\*\* The percentage of reads whose mate pair was also aligned to the reference

## Ib. Tumor Purity

The fraction of reads coming from cross-sample contamination, reflecting a measure of tumor purity, is calculated using GATK's `CalculateContamination` tool.

```{r contamination, echo = F}
# contam_df <- read.delim("tumor_calculatecontamination.table")
# kable_styling(kable(contam_df[, -1],
#                     format = "latex", 
#                     booktabs = TRUE,
#                     align = "c"))
```

\newpage
\blandscape

# II. Germline Alterations

Germline alterations are assessed from the WES sequencing of DNA extracted from blood. Alterations were called using GATK - HaplotypeCaller. Findings were filtered for germline single nucleotide variations (SNVs) and short insertion-deletion events (indels) that were reported as “pathogenic” or “likely pathogenic” in ClinVar`r if(params$type == "glioma") {" (except for the subsection II. f. Common Variants)"}`. Variants of unknown significance (VUS) are not reported. The altered gene, the position of the SNV, the reference nucleotide as well as the altered nucleotide and the mutation type are reported. The read depth at the variation and the predicted allelic frequency are also noted. The report follows a sequential order where a reported positive finding is not reported in the following sections.

### II. a. ACMG Incidental Findings

Any pathogenic/likely-pathogenic germline SNV/indel that affects any of the genes listed in ACMG SF V2.0[^1] are reported undewr this subsection.

[^1]: Kalia SS, Adelman K, Bale SJ, et al. Recommendations for reporting of secondary findings in clinical exome and genome sequencing, 2016 update (ACMG SF v2.0): a policy statement of the American College of Medical Genetics and Genomics. Genet Med. 2017;19(2):249-255.

```{r, ACMG, echo = F, comment=NA}
if(file.exists("./NOTATES/Germline/ACMG.csv")) {
  ACMG_df <- read.csv("./NOTATES/Germline/ACMG.csv")
  rownames(ACMG_df) <- ACMG_df$ID
  ACMG_df <- ACMG_df[,-2]
  
  kable_styling(kable(ACMG_df,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
} else {
  cat("No variation to report.")
}
```

### II. b. Variations in Cancer Gene Census genes

Cancer Gene Census (CGC) from the Catalogue of Somatic Mutations in Cancer (COSMIC) database is a catalogue of genes which contain mutations that have been causally linked to cancer [(https://cancer.sanger.ac.uk/census)](https://cancer.sanger.ac.uk/census). This subsection filters the germline SNV/indels for genes that are reported in CGC. 

```{r, CGC, echo = F, comment=NA}
if(file.exists("./NOTATES/Germline/CGC.csv")) {
  gCGC_df <- read.csv("./NOTATES/Germline/CGC.csv")
  kable_styling(kable(gCGC_df,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
} else {
  cat("No variation to report.")
}
```

### II. c. Variations in Cancer Predisposition Genes

Genes in which germline mutations confer highly or moderately increased risks of cancer are called cancer predisposition genes. Germline SNV/indels catalogued in the 2014 Nature publication by Rahman[^2] are reported in this subsection.

[^2]: Rahman N. Realizing the promise of cancer predisposition genes. Nature. 2014;505(7483):302-8.

```{r, CPG, echo = F, comment=NA}
if(file.exists("./NOTATES/Germline/CPG.csv")) {
  CPG_df <- read.csv("./NOTATES/Germline/CPG.csv")
  kable_styling(kable(CPG_df,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
} else {
  cat("No variation to report.")
}
```

### II. d. Variations in Fanconi Anemia Pathway genes

Fanconi anemia is characterized by the inability to repair DNA interstrand cross-links (ICLs) and results in cancer predisposition[^3]. Genes belonging to this pathway, alterations of which would lead to “BRCA-ness" phenotype are reported here. 

[^3]: Niraj J, Farkkila A, D'andrea AD. The Fanconi Anemia Pathway in Cancer. Annu Rev Cancer Biol. 2019;3:457-478.

```{r, FAP, echo = F, comment=NA}
if(file.exists("./NOTATES/Germline/FAP.csv")) {
  FAP_df <- read.csv("./NOTATES/Germline/FAP.csv")
  kable_styling(kable(FAP_df,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
} else {
  cat("No variation to report.")
}
```

### II. e. Other important germline variations

```{r, OTH, echo = F, comment=NA}
if(file.exists("./NOTATES/Germline/OTHER.csv")) {
  OTH_df <- read.csv("./NOTATES/Germline/OTHER.csv")
  kable_styling(kable(OTH_df,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
} else {
  cat("No variation to report.")
}
```

`r if(params$type == "glioma") {"### II. f. Common Variants"}`

`r if(params$type == "glioma") {"This subsection is distinct from the previous portion of the germline alterations section. Here, germline alterations are filtered for “coding single nucleotide polymorphisms (SNPs)” which were previously shown in genome wide association studies (GWAS) to have an association with gliomas and were listed in the GWAS catalog [(https://www.ebi.ac.uk/gwas/efotraits/EFO_0005543)](https://www.ebi.ac.uk/gwas/efotraits/EFO_0005543)"}`

```{r, common, echo = F, comment=NA}
if (params$type == "glioma") {
  if(file.exists("./NOTATES/Germline/common_var.csv")) {
    common_df <- read.csv("./NOTATES/Germline/common_var.csv")
    kable_styling(kable(common_df,
                        format = "latex", 
                        booktabs = TRUE,
                        align = "c"))
  } else {
    cat("No variation to report.")
  }
}
```

\newpage

# III. Somatic Single Nucleotide Variations (SNVs) and Small Insertion/Deletions(Indels)

### III. a. Tumor Mutational Burden (TMB)

TMB is defined as the number of somatic mutations in the coding region (more accurately the exome capture kit's intervals) per megabase, including single nucleotide variants (SNVs) and small INDELs (insertions and deletions, typically less than 20 bases).

<!-- Others filtering parameters of somatic mutations identified by WES for TMB calculation were a mutated allele frequency greater than 5% and a sequence depth greater than 20X in tumor samples greater or 10X in normal samples. -->

```{r TMB, echo=F, fig.width=12, fig.height=2.5}
somatic_SNVs <- read.delim("./Oncotator/annotated.sSNVs.tsv", comment.char="#")

somatic_SNVs$tot_depth <- vapply(somatic_SNVs$allelic_depth, 
                                 function(x) sum(as.numeric(unlist(strsplit(x, ",")))), 
                                 1, USE.NAMES = FALSE)
somatic_SNVs$lookup <- paste0(somatic_SNVs$Chromosome, somatic_SNVs$Start_position, 
                              somatic_SNVs$End_position, somatic_SNVs$Reference_Allele, 
                              somatic_SNVs$Tumor_Seq_Allele2)

# Filter somatic vars
all_vars <- unique(somatic_SNVs$lookup)
for (s_var in all_vars) {
  tmp <- somatic_SNVs[somatic_SNVs$lookup == s_var, ]
  if (tmp$tot_depth[tmp$Tumor_Sample_Barcode])
}

# subset for (MuTect's default) HQ filters
somatic_SNVs <- subset(somatic_SNVs, 
                       alt_allele_seen=="True")

# subset for tumor_f > 0.05, total depth (tm + normal) > 30X


mut_num <- nrow(somatic_SNVs)
mut_load <- mut_num / as.numeric(params$exome_length)

if (params$type == "glioma") {
  mut_load_df <- read.delim(paste0(params$script_dir, "/mut_loads.txt"), sep = "\t") 
  
  if(!params$ID %in% mut_load_df$Patient) {
    high_conseq <- c("Frame_Shift_Del", "Frame_Shift_Ins", "Splice_Site", 
                 "Translation_Start_Site","Nonsense_Mutation", 
                 "Nonstop_Mutation", "In_Frame_Del","In_Frame_Ins", 
                 "Missense_Mutation")
    somatic_SNVs <- somatic_SNVs[somatic_SNVs$Variant_Classification %in% high_conseq,]
    sset <- ifelse("H3F3A" %in% somatic_SNVs$Hugo_Symbol, "H3-mut",
                   ifelse("IDH1" %in% somatic_SNVs$Hugo_Symbol | "IDH2" %in% somatic_SNVs$Hugo_Symbol, "IDH-mut", "IDH-WT"))
    
    mut_load_df <- rbind(mut_load_df, data.frame(Patient = params$ID, 
                                                 mutational_load = mut_load, 
                                                 subset = sset,
                                                 primary = params$primary_tm))
    write.table(mut_load_df, paste0(params$script_dir, "/mut_loads.txt"), sep = "\t", row.names = F, quote = F)
  }
  
  mut_load_df <- mut_load_df[order(mut_load_df$mutational_load),]
  mut_load_df$x_val <- 1:nrow(mut_load_df)
  
  mut_load_df$for_facet <- ifelse(mut_load_df$primary, "Primary", "Primary + Recurrent")
  tmp <- mut_load_df[mut_load_df$for_facet == "Primary", ]
  tmp$for_facet <- "Primary + Recurrent"
  mut_load_df <- as.data.frame(rbind(mut_load_df, tmp))
  
  suppressPackageStartupMessages(library(ggplot2))
  
  p <- ggplot(mut_load_df, aes(x=x_val, y=mutational_load, group = subset))
  p <- p + geom_point(size=2, aes(shape=subset, color=subset))
  p <- p + scale_color_manual(values=c('#56B4E9','#E69F00', 'gray60'))
  p <- p + geom_segment(aes(x = median(mut_load_df$x_val)-6, y = median(mut_load_df$mutational_load), 
                            xend = median(mut_load_df$x_val)+6, yend = median(mut_load_df$mutational_load)), colour = "red")
  p <- p + coord_flip()
  p <- p + theme(axis.title.y=element_blank(),
                 axis.text.y=element_blank(),
                 axis.ticks.y=element_blank())
  p <- p + scale_y_log10(breaks = scales::trans_breaks("log10", function(x) 10^x),
                         labels = scales::trans_format("log10", scales::math_format(10^.x)))
  p <- p + ylab("Mutational Load")
  p <- p + geom_point(data=mut_load_df[mut_load_df$Patient == params$ID, ], aes(x=x_val, y=mutational_load), colour="red", size=4, shape = 1)
  p <- p + facet_grid(rows = vars(for_facet))
  p
}
```

`r if(params$type == "glioma") {paste0("Mutational loads (Numbers of mutations per Mb) for ", nrow(mut_load_df), " patients (including the current patient, circled in red) are represented in the above scatter plot. Red vertical line segment represents the overall median mutational load value.")}`

```{r load_table, echo = F, comment=NA}
if (params$type == "glioma") {
  load_tbl_ALL <- tapply(mut_load_df$mutational_load, mut_load_df$subset, median)
  load_tbl_ALL <- as.data.frame(load_tbl_ALL)
  load_tbl_ALL <- rbind(load_tbl_ALL, median(mut_load_df$mutational_load))
  rownames(load_tbl_ALL)[nrow(load_tbl_ALL)] <- "Overall"
  colnames(load_tbl_ALL) <- "Primary (untreated) + Recurrent (previoulsy treated)"
  
  tmp <- mut_load_df[mut_load_df$primary, ]
  load_tbl_primary <- tapply(tmp$mutational_load, tmp$subset, median)
  load_tbl_primary <- as.data.frame(load_tbl_primary)
  load_tbl_primary <- rbind(load_tbl_primary, median(tmp$mutational_load))
  rownames(load_tbl_primary)[nrow(load_tbl_primary)] <- "Overall"
  colnames(load_tbl_primary) <- "Primary (untreated)"
  
  load_tbl <- cbind(load_tbl_ALL, load_tbl_primary)
  
  kable_styling(kable(load_tbl,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
}
```

** Mutation load (Number of mutations per Mb) for the current patient is: `r mut_load` **

### III. b. Microsatellite Instability Status

```{r classification, echo = F, comment=NA}
class_tbl <- read.csv("MSIpred/MSIpred_prediction.csv", row.names = 1)
rownames(class_tbl) <- NULL
kable_styling(kable(class_tbl,
                    format = "latex", booktabs = TRUE), 
              latex_options = "scale_down")
```

`r if(params$type == "glioma") {"### III. c. Important SNVs in Glioma"} else {"### III. c. Important SNVs in Medulloblastoma"}`

`r if(params$type == "glioma") {"This is the intersection of SNVs in this case with a list of SNVs manually curated because of their importance in gliomas."} else {"This is the intersection of SNVs in this case with a list of SNVs manually curated because of their importance in medulloblastomas."}`


```{r imp_snv, echo = F, comment=NA}
if(file.exists("./NOTATES/Somatic_SNV/important_SNVs.csv"))
{
  imp_snv <- read.csv("./NOTATES/Somatic_SNV/important_SNVs.csv", colClasses = c("character"))
  imp_snv <- imp_snv[,setdiff(colnames(imp_snv),c("COSMIC_total_alterations_in_gene","DNA_repair","selected_KEGG"))]
  colnames(imp_snv) <- c("Gene", "Effect", "Protein", "Genome","TumorF","Genotype", "COS_n_ov",
                        "UniProt_Region", "SIFT","Irino", "TMZ")
  
  kable_styling(kable(imp_snv,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
}else
{
  cat("No variation to report.")
}
```

### III. d. CGC gene + COSMIC hotspot

These are the somatic SNVs/indels in this case in a CGC gene, positioned at a site with at least one COSMIC overlapping mutation.

```{r, hotspot, echo = F, comment=NA}
if(file.exists("./NOTATES/Somatic_SNV/COSMIC_hotspot.csv"))
{
  hot_snv <- read.csv("./NOTATES/Somatic_SNV/COSMIC_hotspot.csv", colClasses = c("character"))
  hot_snv <- hot_snv[,setdiff(colnames(hot_snv),c("COSMIC_total_alterations_in_gene","DNA_repair","selected_KEGG"))]
  colnames(hot_snv) <- c("Gene", "Effect", "Protein", "Genome","TumorF","Genotype", "COS_n_ov",
                        "UniProt_Region", "SIFT", "Irino", "TMZ")
  kable_styling(kable(hot_snv,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
}else
{
  cat("No variation to report.")
}
```

### III. e. CGC gene only

These are the SNVs in a CGC gene, not positioned at a site with any COSMIC overlapping mutations.

```{r, cgc_snv, echo = F, comment=NA}
if(file.exists("./NOTATES/Somatic_SNV/CGC_genes.csv"))
{
  cgc_snv <- read.csv("./NOTATES/Somatic_SNV/CGC_genes.csv", colClasses = c("character"))
  cgc_snv <- cgc_snv[,setdiff(colnames(cgc_snv),c("COSMIC_total_alterations_in_gene","DNA_repair","selected_KEGG"))]
  colnames(cgc_snv) <- c("Gene", "Effect", "Protein", "Genome","TumorF","Genotype", "COS_n_ov",
                        "UniProt_Region", "SIFT", "Irino", "TMZ")
    kable_styling(kable(cgc_snv,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
} else
{
  cat("No variation to report.")
}
```

### III. f. Other possibly important SNVs

SNVs in DNA damage repair genes:

```{r, DDR, echo = F, comment=NA}
if(file.exists("./NOTATES/Somatic_SNV/DDR_related.csv"))
{
  ddr_snv <- read.csv("./NOTATES/Somatic_SNV/DDR_related.csv", colClasses = c("character"))

  ddr_snv <- ddr_snv[,setdiff(colnames(ddr_snv),c("COSMIC_total_alterations_in_gene","selected_KEGG"))]
  colnames(ddr_snv) <- c("Gene", "Effect", "Protein", "Genome","TumorF","Genotype", "COS_n_ov",
                        "UniProt_Region", "SIFT", "DNA_repair", "Irino", "TMZ")
  kable_styling(kable(ddr_snv,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
}else
{
  cat("No variation to report.")
}
```

SNVs in important KEGG Pathway genes:

```{r, kegg, echo = F, comment=NA}
if(file.exists("./NOTATES/Somatic_SNV/KEGG_selected.csv")) {
  kegg_snv <- read.csv("./NOTATES/Somatic_SNV/KEGG_selected.csv", colClasses = c("character"))

  colnames(kegg_snv) <- c("Gene", "Effect", "Protein", "Genome","TumorF","Genotype", "COS_n_ov",
                          "COS_tot", "UniProt_Region", "SIFT", "DNA_repair", "Irino", "TMZ", "KEGG")

  kable_styling(kable(kegg_snv,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
} else {
  cat("No variation to report.")
}
```

SNVs in important Irinotecan response genes:

```{r, irino, echo = F, comment=NA}
if(file.exists("./NOTATES/Somatic_SNV/irinotecan.csv")) {
  irino_snv <- read.csv("./NOTATES/Somatic_SNV/irinotecan.csv", colClasses = c("character"))

  colnames(irino_snv) <- c("Gene", "Effect", "Protein", "Genome","TumorF","Genotype", "COS_n_ov",
                          "COS_tot", "UniProt_Region", "SIFT", "DNA_repair", "Irino", "TMZ", "KEGG")
  
  kable_styling(kable(iirino_snv,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
} else {
  cat("No variation to report.")
}
```

SNVs in important Temozolomide resistance genes:

```{r, tmz, echo = F, comment=NA}
if(file.exists("./NOTATES/Somatic_SNV/tmz_res.csv")) {
  tmz_snv <- read.csv("./NOTATES/Somatic_SNV/tmz_res.csv", colClasses = c("character"))
  
  colnames(tmz_snv) <- c("Gene", "Effect", "Protein", "Genome","TumorF","Genotype", "COS_n_ov",
                          "COS_tot", "UniProt_Region", "SIFT", "DNA_repair", "Irino", "TMZ", "KEGG")

  kable_styling(kable(tmz_snv,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
} else {
  cat("No variation to report.")
}
```

Other coding mutations and non-coding mutations are provided as attachments.

\newpage

# IV. Somatic Copy Number Alterations (SCNAs)

Only SCNAs with a $log_2(Tumor/Normal)$ ratio $\le$ -0.25 or $\ge$ 0.2 are reported in the tables.

`r if(params$type == "glioma") {"### IV. a. Important SCNAs in Glioma"} else {"### IV. a. Important SCNAs in Medulloblastoma"}`

`r if(params$type == "glioma") {"This is the intersection of SCNAs in this case with a list of SCNAs manually curated because of their importance in gliomas."} else {"This is the intersection of SCNAs in this case with a list of SCNAs manually curated because of their importance in medulloblastomas."}`

```{r, imp_cna, echo = F, comment=NA}
if(file.exists("./NOTATES/SCNA/important_CNVs.csv")) {
  imp_cna <- read.csv("./NOTATES/SCNA/important_CNVs.csv")
  kable_styling(kable(imp_cna,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
} else {
  cat("No variation to report.")
}
```

### IV. b. SCNAs in Cancer Gene Census genes

```{r, cgc_cna, echo = F, comment=NA}
if(file.exists("./NOTATES/SCNA/CGC_CNVs.csv")) {
  cgc_cna <- read.csv("./NOTATES/SCNA/CGC_CNVs.csv")
  kable_styling(kable(cgc_cna[,-c(2,3)],
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
} else {
  cat("No variation to report.")
}
```

### IV. c. Broad SCNAs

```{r, brod_cna, echo = F, comment=NA}
if(file.exists("./NOTATES/SCNA/broad.csv")) {
  broad_cna <- read.csv("./NOTATES/SCNA/broad.csv")
  kable_styling(kable(broad_cna,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
} else {
  cat("No variation to report.")
}
```

### IV. d. SCNA by Chromosome Plots

\begin{center}
\includegraphics[width=600pt]{./ExomeCNV/{CNV.cnv}.png}
\end{center}

\newpage

# V. Loss of Heterozygosity (LOH) Events

Only LOH events for which absolute difference of B-allele frequencies ($|BAF_{Tumor}-BAF_{Normal}|$) is larger than 0.4 are reported.

### V. a. LOH Overview

```{r, loh_regions, echo = F, comment=NA}
loh <- read.csv("./NOTATES/LOH/LOH.csv")
loh$Position <- paste0(loh$Chr, ":", loh$Start, "-", loh$End)
loh <- loh[,c("Genes", "Position", "N_BAF", "T_BAF", "Absolute_Diff")]

kable_styling(kable(loh,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
```

### V. b. LOH + somatic SNV/Indel

```{r, loh_dhit, echo = F, comment=NA}
if(file.exists("./NOTATES/LOH/d_hit_loh.csv"))
{
  loh_dhit <- read.csv("./NOTATES/LOH/d_hit_loh.csv")
  kable_styling(kable(loh_dhit,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
}else
{
  cat("No event to report.")
}
```


### V. c. LOH Events in CGC genes 

```{r, loh_cgc, echo = F, comment=NA}
if(file.exists("./NOTATES/LOH/CGC_loh.csv"))
{
  loh_cgc <- read.csv("./NOTATES/LOH/CGC_loh.csv")
  kable_styling(kable(loh_cgc,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
}else
{
  cat("No event to report.")
}
```

\newpage

# VI. Tumor Heterogeneity Analysis 
```{r n3_or_n2, echo=F}
if(!file.exists("./THetA/output/CNV.BEST.results")){
   n3_chosen <- F
}else{
  theta_out <- read.delim("./THetA/output/CNV.BEST.results")
  n3_chosen <- ifelse(length(unlist(strsplit(theta_out[1,"mu"], ","))) == 2, F, T)
}
```

THetA (Tumor Heterogeneity Analysis) is an algorithm that estimates the tumor purity and clonal/sublconal copy number aberrations directly from high-throughput DNA sequencing data.

\begin{center} 
\includegraphics[width=500pt]{`r ifelse(n3_chosen,"./THetA/output/CNV.n3.graph.pdf","./THetA/output/CNV.n2.graph.pdf")`}
\end{center}

\elandscape
\newpage

# VII. Mutational Signatures

\begin{center} 
\includegraphics[width=500pt]{mut_signatures.pdf}
\end{center}

\begin{center} 
\includegraphics[width=400pt]{mut_signatures_pie.pdf}
\end{center}

\newpage
\blandscape

```{r, mut_sigs, echo = F, comment=NA}
mut_sigs <- read.csv("./mut_signatures.csv", row.names = 1)
kable_styling(kable(mut_sigs,
                    format = "latex", booktabs = TRUE),
              latex_options = "scale_down")
```

\newpage

# VIII. KEGG Pathway Enrichment Analysis

Genes that harbor non-synonymous exonic mutations and/or homozygous deletion or multi-copy amplification (3+) are included in the enrichment analysis.

### VIII. a. KEGG - Pathways in Cancer

\begin{center}
\includegraphics[width=500pt]{./pathways/{hsa05200.CANCER.multi}.png}
\end{center}

The right half of a gene indicates copy number ratio (scaled between -1 and 1) while the left half indicates somatic mutation status (1 if mutant).

### VIII. b. KEGG - Pathway Enrichment Results

```{r, enr_res, echo = F, comment=NA}
enr_res <- read.csv("pathways/enrichment_results.csv", row.names = 1)
kable_styling(kable(enr_res,
                    format = "latex", booktabs = TRUE), 
              latex_options = "scale_down")
```

\elandscape
