---
output: pdf_document
params:
  ID: ""
  script_dir: ""
  exome_length: ""
  exome_bed: ""
  type: ""
  primary_tm: ""
  tumor_sample: ""
header-includes:
- \usepackage{titling}
- \pretitle{\begin{center}\LARGE\includegraphics[width=160pt]{`r file.path(params$script_dir, "NOT_logo.png")`}\\[\bigskipamount]}
- \posttitle{\end{center}}
- \usepackage{float}
- \usepackage{booktabs}
- \usepackage[table]{xcolor}
title: \ \ NeuroOncology Technologies (NOT)
author: "Whole Exome Sequencing Report"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, comment = NA)
options(stringsAsFactors = FALSE, Encoding="UTF-8")
suppressPackageStartupMessages(library(knitr))
suppressPackageStartupMessages(library(kableExtra))
suppressPackageStartupMessages(library(GenomicRanges))
```

\pagenumbering{gobble} 
\vspace{2cm}

***

>>> `r paste("**NOT ID: **", params$ID)` \
**Treatment status:** `r ifelse(as.logical(params$primary_tm), "Primary (untreated) tumor", "Recurrent (previously treated) tumor")` \
**Tumor sample:** `r ifelse(params$tumor_sample == "FFPE", "Formalin-Fixed Paraffin-Embedded (FFPE) tissue specimen", "Fresh tumor tissue, snap-frozen in liquid nitrogen")` \
**Normal sample:** Peripheral venous blood \
**DNA extraction method:** QIAGEN DNeasy Blood & Tissue kit \
**Report date:** `r format(Sys.time(), '%d %B, %Y')`

***


\raggedright
\clearpage
\pagenumbering{arabic}


# I. Quality Metrics

## Ia. Summary Table of Quality Metrics

```{r QC}
QC_df <- read.csv("./QC_table.csv" )
colnames(QC_df)[1] <- ""
kable_styling(kable(QC_df,
                    format = "latex", 
                    booktabs = TRUE,
                    align = c("r", "c", "c")), 
              latex_options = "striped",
              stripe_color = "lightgray")
```

>>> \* PF is defined as passing Illumina's filter \
    \*\* The percentage of PF reads that aligned to the reference sequence \
    \*\*\* The percentage of reads whose mate pair was also aligned to the reference

## Ib. Tumor Purity

The fraction of reads coming from cross-sample contamination, reflecting a measure of tumor purity, is calculated using the GATK - `CalculateContamination` tool. This tool calculates the fraction of reads coming from cross-sample contamination and estimates contamination based on the signal from reference reads at homozygous alternate sites. Another purity/clonality estimate based on copy number alterations (CNAs) is presented under section VII. Tumor Heterogeneity Analysis.

```{r contamination}
contam_df <- read.delim("contamination.table")
kable_styling(kable(contam_df[, -1],
                    format = "latex",
                    booktabs = TRUE,
                    align = "c"))
```

\newpage

# II. Germline Alterations

Germline alterations are assessed from the WES sequencing of DNA extracted from blood. Alterations were called using GATK - HaplotypeCaller. Findings were filtered for germline single nucleotide variations (SNVs) and short insertion-deletion events (indels) that were reported as “pathogenic” or “likely pathogenic” in ClinVar`r if(params$type == "glioma") {" (except for the subsection II. f. Common Variants)"}`. Variants of unknown significance (VUS) are not reported. The altered gene, the position of the SNV, the reference nucleotide as well as the altered nucleotide and the mutation type are reported. The read depth at the variation and the predicted allelic frequency are also noted. The report follows a sequential order where a reported positive finding is not reported in the following sections.

### II. a. ACMG Incidental Findings

Any pathogenic/likely-pathogenic germline SNV/indel that affects any of the genes listed in ACMG SF V2.0[^1] are reported undewr this subsection.

[^1]: Kalia SS, Adelman K, Bale SJ, et al. Recommendations for reporting of secondary findings in clinical exome and genome sequencing, 2016 update (ACMG SF v2.0): a policy statement of the American College of Medical Genetics and Genomics. Genet Med. 2017;19(2):249-255.

```{r ACMG}
if(file.exists("./NOTATES/Germline/ACMG.csv")) {
  ACMG_df <- read.csv("./NOTATES/Germline/ACMG.csv")
  rownames(ACMG_df) <- ACMG_df$ID
  ACMG_df <- ACMG_df[,-2]
  
  kable_styling(kable(ACMG_df,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
} else {
  cat("No variation to report.")
}
```

### II. b. Variations in Cancer Gene Census genes

Cancer Gene Census (CGC) from the Catalogue of Somatic Mutations in Cancer (COSMIC) database is a catalogue of genes which contain mutations that have been causally linked to cancer [(https://cancer.sanger.ac.uk/census)](https://cancer.sanger.ac.uk/census). This subsection filters the germline SNV/indels for genes that are reported in CGC. 

```{r CGC}
if(file.exists("./NOTATES/Germline/CGC.csv")) {
  gCGC_df <- read.csv("./NOTATES/Germline/CGC.csv")
  kable_styling(kable(gCGC_df,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
} else {
  cat("No variation to report.")
}
```

### II. c. Variations in Cancer Predisposition Genes

Genes in which germline mutations confer highly or moderately increased risks of cancer are called cancer predisposition genes. Germline SNV/indels catalogued in the 2014 Nature publication by Rahman[^2] are reported in this subsection.


[^2]: Rahman N. Realizing the promise of cancer predisposition genes. Nature. 2014;505(7483):302-8.

```{r CPG}
if(file.exists("./NOTATES/Germline/CPG.csv")) {
  CPG_df <- read.csv("./NOTATES/Germline/CPG.csv")
  kable_styling(kable(CPG_df,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
} else {
  cat("No variation to report.")
}
```


### II. d. Variations in Fanconi Anemia Pathway genes

Fanconi anemia is characterized by the inability to repair DNA interstrand cross-links (ICLs) and results in cancer predisposition[^3]. Genes belonging to this pathway, alterations of which would lead to “BRCA-ness" phenotype are reported here. 

[^3]: Niraj J, Farkkila A, D'andrea AD. The Fanconi Anemia Pathway in Cancer. Annu Rev Cancer Biol. 2019;3:457-478.

```{r FAP}
if(file.exists("./NOTATES/Germline/FAP.csv")) {
  FAP_df <- read.csv("./NOTATES/Germline/FAP.csv")
  kable_styling(kable(FAP_df,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
} else {
  cat("No variation to report.")
}
```

### II. e. Other Important Germline Variations

Germline variations in other manually curated genes (alterations of which may carry clinical importance) are reported under this subsection.

```{r OTH}
if(file.exists("./NOTATES/Germline/OTHER.csv")) {
  OTH_df <- read.csv("./NOTATES/Germline/OTHER.csv")
  kable_styling(kable(OTH_df,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
} else {
  cat("No variation to report.")
}
```

`r if(params$type == "glioma") {"### II. f. Common Variants"}`

`r if(params$type == "glioma") {"This subsection is distinct from the previous portion of the germline alterations section. Here, germline alterations are filtered for “coding single nucleotide polymorphisms (SNPs)” which were previously shown in genome wide association studies (GWAS) to have an association with gliomas and were listed in the GWAS catalog [(https://www.ebi.ac.uk/gwas/efotraits/EFO_0005543)](https://www.ebi.ac.uk/gwas/efotraits/EFO_0005543)"}`

```{r common}
if (params$type == "glioma") {
  if(file.exists("./NOTATES/Germline/common_var.csv")) {
    common_df <- read.csv("./NOTATES/Germline/common_var.csv")
    kable_styling(kable(common_df,
                        format = "latex", 
                        booktabs = TRUE,
                        align = "c"))
  } else {
    cat("No variation to report.")
  }
}
```

\newpage

# III. Somatic Single Nucleotide Variations (SNVs) and Small Insertion/Deletions(Indels)

Somatic mutations were called from the tumor-blood pair in the current analysis using GATK - MuTect2 with a cutoff of 5% variant allele frequency (VAF). Only non-synonymous variants (with variant classifications of High/Moderate variant consequences: "Frame_Shift_Del", "Frame_Shift_Ins", "Splice_Site", "Translation_Start_Site","Nonsense_Mutation", "Nonstop_Mutation", "In_Frame_Del","In_Frame_Ins", "Missense_Mutation", see [http://www.ensembl.org/Help/Glossary?id=535]( http://www.ensembl.org/Help/Glossary?id=535)) were used for reporting. Genes that are often non-pathogenic and passengers but are frequently mutated in most of the public exome studies (named FLAGS) as collected by Shyr et al. were excluded from the report [^4]. 

[^4]: Shyr C, Tarailo-graovac M, Gottlieb M, Lee JJ, Van karnebeek C, Wasserman WW. FLAGS, frequently mutated genes in public exomes. BMC Med Genomics. 2014;7:64.

### III. a. Tumor Mutational Burden (TMB)

TMB is defined as the number of somatic mutations in the coding region (more accurately the exome capture kit's intervals) per megabase, including single nucleotide variants (SNVs) and small indels (insertions and deletions, typically less than 20 bases).

This calculation was performed using the following filtering approach:

1. keeping variants with mutated allele frequency greater than 5%
2. keeping variants with a sequence depth greater than 20X in the tumor sample greater and 10X in the normal sample

```{r TMB, fig.width=10, fig.height=3}
somatic_SNVs <- read.delim("./Oncotator/annotated.sSNVs.tsv", comment.char="#")

somatic_SNVs$tot_depth <- vapply(somatic_SNVs$allelic_depth, 
                                 function(x) sum(as.numeric(unlist(strsplit(x, ",")))), 
                                 1, USE.NAMES = FALSE)

# Filter for depth greater than 20X in tumor samples greater or 10X in normal samples
thresholds <- ifelse(somatic_SNVs$alt_allele_seen == "True", 20, 10)
somatic_SNVs$dep_keep <- somatic_SNVs$tot_depth > thresholds
tmp <- somatic_SNVs[somatic_SNVs$alt_allele_seen == "False" & !somatic_SNVs$dep_keep, ]
somatic_SNVs$dep_keep[somatic_SNVs$Genome_Change %in% tmp$Genome_Change] <- FALSE
somatic_SNVs <- somatic_SNVs[somatic_SNVs$dep_keep, ]

somatic_SNVs <- somatic_SNVs[somatic_SNVs$alt_allele_seen == "True", ]

# subset for tumor_f > 0.05
somatic_SNVs <- somatic_SNVs[somatic_SNVs$tumor_f > 0.05, ]

## Calculate TMB
mut_num <- nrow(somatic_SNVs)
current_TMB <- mut_num / as.numeric(params$exome_length)

TMB_tbl_path <- file.path(params$script_dir, 
                          paste0(params$type, "_TMB_table.RDS"))
if (!file.exists(TMB_tbl_path)) {
  TMB_df <- data.frame()
} else {
  TMB_df <- readRDS(TMB_tbl_path)
}

## add to TMB table if not present
if(!params$ID %in% TMB_df$ID) {
  high_conseq <- c("Frame_Shift_Del", "Frame_Shift_Ins", "Splice_Site", 
                   "Translation_Start_Site","Nonsense_Mutation", 
                   "Nonstop_Mutation", "In_Frame_Del","In_Frame_Ins", 
                   "Missense_Mutation")
  somatic_SNVs <- somatic_SNVs[somatic_SNVs$Variant_Classification %in% high_conseq,]
  current_sset <- ifelse("H3F3A" %in% somatic_SNVs$Hugo_Symbol, "H3-mut",
                         ifelse("IDH1" %in% somatic_SNVs$Hugo_Symbol | "IDH2" %in% somatic_SNVs$Hugo_Symbol, "IDH-mut", "IDH-WT"))
  
  TMB_df <- rbind(TMB_df, 
                  data.frame(ID = params$ID, 
                             TMB = current_TMB, 
                             subset = current_sset,
                             primary = params$primary_tm))
  saveRDS(TMB_df, TMB_tbl_path)
}

## Scatter plots of TMB over all analyses
# rank TMB vals
TMB_df <- TMB_df[order(TMB_df$TMB),]
TMB_df$x_val <- 1:nrow(TMB_df)

# prep for facets
TMB_df$for_facet <- ifelse(TMB_df$primary, "Primary", "Primary\n+\nRecurrent")
pr_tms <- TMB_df[TMB_df$for_facet == "Primary", ]
pr_tms$for_facet <- "Primary\n+\nRecurrent"
TMB_df_plot <- as.data.frame(rbind(TMB_df, pr_tms))

# calculate height of median line segment
x_shift <- max(nrow(TMB_df_plot) / 7, 3)

suppressPackageStartupMessages(library(ggplot2))
p <- ggplot(TMB_df_plot, aes(x=x_val, y=TMB, group = subset))
p <- p + geom_point(size=2, aes(shape=subset, color=subset))
p <- p + scale_color_manual(values=c('#56B4E9','#E69F00', 'gray60'))
p <- p + geom_segment(aes(x = median(TMB_df_plot$x_val) - x_shift, 
                          y = median(TMB_df_plot$TMB), 
                          xend = median(TMB_df_plot$x_val) + x_shift, 
                          yend = median(TMB_df_plot$TMB)), 
                      colour = "red")
p <- p + coord_flip()
p <- p + theme(axis.title.y=element_blank(),
               axis.text.y=element_blank(),
               axis.ticks.y=element_blank())
p <- p + scale_y_log10(breaks = scales::trans_breaks("log10", function(x) 10^x),
                       labels = scales::trans_format("log10", scales::math_format(10^.x)))
p <- p + ylab("Mutational Load")
p <- p + geom_point(data=TMB_df_plot[TMB_df_plot$ID == params$ID, ], 
                    aes(x=x_val, y=TMB), colour="red", size=4, shape = 1)
p <- p + facet_grid(rows = vars(for_facet))
p
```

`r if(params$type == "glioma") {paste0("TMBs (mutations/Mb) of all previously reported tumors, including the current tumor, circled in red are presented in the above scatter plots, separately displayed for (a) primary (untreated tumors only, top panel, n =", nrow(pr_tms), ") and (b) primary + recurrent tumors (untreated tumors and tumors previously undergone treatment, bottom panel, n=", nrow(TMB_df), "). The red vertical lines indicate median TMB for each panel. The table below displays the median TMBs per glioma molecular subsets over all previous analyses for primary and primary+recurrent tumors separately.")}`

```{r load_table}
if (params$type == "glioma") {
  ss_TMB_tbl_ALL <- tapply(TMB_df$TMB, TMB_df$subset, median)
  ss_TMB_tbl_ALL <- as.data.frame(ss_TMB_tbl_ALL)
  ss_TMB_tbl_ALL <- rbind(ss_TMB_tbl_ALL, 
                          median(TMB_df$TMB))
  rownames(ss_TMB_tbl_ALL)[nrow(ss_TMB_tbl_ALL)] <- "Overall"
  colnames(ss_TMB_tbl_ALL) <- "Primary\n+\nRecurrent"
  
  tmp <- TMB_df[TMB_df$primary, ]
  ss_TMB_tbl_prim <- tapply(tmp$TMB, tmp$subset, median)
  ss_TMB_tbl_prim <- as.data.frame(ss_TMB_tbl_prim)
  ss_TMB_tbl_prim <- rbind(ss_TMB_tbl_prim, 
                           median(tmp$TMB))
  rownames(ss_TMB_tbl_prim)[nrow(ss_TMB_tbl_prim)] <- "Overall"
  colnames(ss_TMB_tbl_prim) <- "Primary"
  
  ss_TMB_tbl <- cbind(ss_TMB_tbl_ALL, ss_TMB_tbl_prim)
  
  kable_styling(kable(ss_TMB_tbl,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
}
```

> Mutation load (Number of mutations per Mb) for the current tumor sample is: `r current_TMB` 

### III. b. Microsatellite Instability Status

The microsatellite instability (MSI) status was predicted using the tool MSIpred [5]. MSIpred computes 22 somatic mutational features (predictors of MSI) and using these features, it predicts tumor MSI status as microsatellite instability high (MSI-H) or microsatellite stable (MSS) using a machine learning method, support vector machine (SVM). 

Additionally, polymerase-epsilon (POLE) deficiency is predicted based on the presence of both of:

1. total count of somatic SNVs / Mb > 60 AND
2. count of somatic indels in single sequence repats / Mb < 0.18

[5]: Wang C, Liang C. MSIpred: a python package for tumor microsatellite instability classification from tumor mutation annotation data using a support vector machine. Sci Rep. 2018;8(1):17546.


```{r classification}
class_tbl <- read.csv("MSIpred/MSIpred_prediction.csv", row.names = 1)
rownames(class_tbl) <- NULL
kable_styling(kable(class_tbl,
                    format = "latex", 
                    booktabs = TRUE,
                    align = "c"))
```

`r if(params$type == "glioma") {"### III. c. Important SNV/indels in Glioma"} else {"### III. c. Important  SNV/indels in Medulloblastoma"}`

`r if(params$type == "glioma") {"This subsection contains somatic SNV/indels in genes that have been reported in the TCGA pan-glioma study of Ceccarelli et al. [^5] which alayzed 1122 WHO grade II-III and IV diffuse-gliomas."} else {"This is the intersection of somaticSNV/indels in this case with a list of genes manually curated because of their importance in medulloblastomas."}`

`r if(params$type == "glioma") {"[^5]: Ceccarelli M, Barthel FP, Malta TM, et al. Molecular Profiling Reveals Biologically Discrete Subsets and Pathways of Progression in Diffuse Glioma. Cell. 2016;164(3):550-63."}`


```{r imp_som_var}
if(file.exists("./NOTATES/Somatic_SNV/important_SNVs.csv")) {
  imp_som_var <- read.csv("./NOTATES/Somatic_SNV/important_SNVs.csv", colClasses = c("character"))
  imp_som_var <- imp_som_var[, setdiff(colnames(imp_som_var), 
                               c("COSMIC_total_alterations_in_gene", "DNA_repair", "selected_KEGG", "genotype"))]
  colnames(imp_som_var) <- c("Gene", "Variant Classification", "Protein", "Genomic Change", "VAF", "COS_n_ov",
                         "UniProt_Region", "SIFT", "Irino", "TMZ")
  
  k_tbl <- kable(imp_som_var, format = "latex", booktabs = TRUE)
  k_tbl <- column_spec(k_tbl, which(colnames(imp_som_var) == "UniProt_Region"), width = "10em")
  kable_styling(k_tbl, latex_options = "scale_down")
} else {
  cat("No variation to report.")
}
```

### III. d. CGC gene + COSMIC hotspot

This subsection presents somatic SNV/indels which fulfill the following criteria:

1. The gene harboring the variants is listed in the Catalogue of Somatic Mutations in Cancer (COSMIC) - CGC, which is a catalogue of genes which contain mutations that have been causally linked to cancer [(https://cancer.sanger.ac.uk/census)](https://cancer.sanger.ac.uk/census). 
2. The variant has a mutation record in COSMIC at the same genomic position (as a recurrent mutation increases the likelihood of an actionable “gain-of function” mutation).

```{r hotspot}
if(file.exists("./NOTATES/Somatic_SNV/COSMIC_hotspot.csv")) {
  hot_som <- read.csv("./NOTATES/Somatic_SNV/COSMIC_hotspot.csv", colClasses = c("character"))
  hot_som <- hot_som[, setdiff(colnames(hot_som), 
                               c("COSMIC_total_alterations_in_gene", "DNA_repair", "selected_KEGG", "genotype"))]
  colnames(hot_som) <- c("Gene", "Variant Classification", "Protein", "Genomic Change", "VAF", "COS_n_ov",
                         "UniProt_Region", "SIFT","Irino", "TMZ")
  k_tbl <- kable(hot_som, format = "latex", booktabs = TRUE)
  k_tbl <- column_spec(k_tbl, which(colnames(hot_som) == "UniProt_Region"), width = "10em")
  kable_styling(k_tbl, latex_options = "scale_down")
} else {
  cat("No variation to report.")
}
```

### III. e. CGC gene only

This subsection lists somatic variants where the gene harboring the variants is listed in the Catalogue of Somatic Mutations in Cancer (COSMIC) - CGC, which is a catalogue of genes which contain mutations that have been causally linked to cancer [(https://cancer.sanger.ac.uk/census)](https://cancer.sanger.ac.uk/census).

```{r cgc_som}
if(file.exists("./NOTATES/Somatic_SNV/CGC_genes.csv")) {
  cgc_som <- read.csv("./NOTATES/Somatic_SNV/CGC_genes.csv", colClasses = c("character"))
  cgc_som <- cgc_som[, setdiff(colnames(cgc_som), 
                               c("COSMIC_total_alterations_in_gene", "DNA_repair", "selected_KEGG", "genotype"))]
  colnames(cgc_som) <- c("Gene", "Variant Classification", "Protein", "Genomic Change","VAF", "COS_n_ov",
                         "UniProt_Region", "SIFT","Irino", "TMZ")
  k_tbl <- kable(cgc_som, format = "latex", booktabs = TRUE)
  k_tbl <- column_spec(k_tbl, which(colnames(cgc_som) == "UniProt_Region"), width = "10em")
  kable_styling(k_tbl, latex_options = "scale_down")
} else
{
  cat("No variation to report.")
}
```

### III. f. Other possibly important somatic SNV/indels

Somatic SNV/indels in DNA damage repair genes that were presented on [https://www.mdanderson.org/documents/Labs/Wood-Laboratory/human-dna-repair-genes.html](https://www.mdanderson.org/documents/Labs/Wood-Laboratory/human-dna-repair-genes.html):

```{r DDR}
if(file.exists("./NOTATES/Somatic_SNV/DDR_related.csv")) {
  ddr_som_var <- read.csv("./NOTATES/Somatic_SNV/DDR_related.csv", colClasses = c("character"))

  ddr_som_var <- ddr_som_var[, setdiff(colnames(ddr_som_var),
                                       c("COSMIC_total_alterations_in_gene", "selected_KEGG", "genotype"))]
  colnames(ddr_som_var) <- c("Gene", "Variant Classification", "Protein", "Genomic Change", "VAF", "COS_n_ov",
                             "UniProt_Region", "SIFT", "DNA_repair", "Irino", "TMZ")
  k_tbl <- kable(ddr_som_var, format = "latex", booktabs = TRUE)
  k_tbl <- column_spec(k_tbl, which(colnames(ddr_som_var) == "UniProt_Region"), width = "10em")
  kable_styling(k_tbl, latex_options = "scale_down")
} else {
  cat("No variation to report.")
}
```

Somatic SNVs/indels in important KEGG Pathway genes:

```{r kegg}
if(file.exists("./NOTATES/Somatic_SNV/KEGG_selected.csv")) {
  kegg_som_var <- read.csv("./NOTATES/Somatic_SNV/KEGG_selected.csv", colClasses = c("character"))
  kegg_som_var <- kegg_som_var[, colnames(kegg_som_var) != "genotype"]

  colnames(kegg_som_var) <- c("Gene", "Variant Classification", "Protein", "Genomic Change", "VAF", "COS_n_ov",
                              "COS_tot", "UniProt_Region", "SIFT", "DNA_repair", "Irino", "TMZ", "KEGG")
  
  k_tbl <- kable(kegg_som_var, format = "latex", booktabs = TRUE)
  k_tbl <- column_spec(k_tbl, which(colnames(kegg_som_var) == "UniProt_Region"), width = "10em")
  kable_styling(k_tbl, latex_options = "scale_down")
} else {
  cat("No variation to report.")
}
```

\newpage

# IV. Somatic Copy Number Alterations (SCNAs)

Only SCNAs with a $log_2(Tumor/Normal)$ ratio $\le$ -0.25 or $\ge$ 0.2 were used in analysis.

### IV. a. SCNA Burden

Below are 4 metrics associated with SCNA Burden:

```{r SCNA_burden}
######### intersect SCNA segments with exome segments
segs_df <- read.csv("NOTATES/SCNA/exomeCNV.csv")
segs_gr <- makeGRangesFromDataFrame(segs_df, 
                                    seqnames.field = "chr", 
                                    start.field = "probe_start", 
                                    end.field = "probe_end")
exome_df <- read.delim(params$exome_bed, header = FALSE)
exome_gr <- makeGRangesFromDataFrame(exome_df, 
                                     seqnames.field = "V1", 
                                     start.field = "V2", 
                                     end.field = "V3")
instersection_gr <- intersect(segs_gr, exome_gr)

######### Calculate SCNA burden metrics
intersection_df <- as.data.frame(instersection_gr)
intersection_df$len <- intersection_df$end - intersection_df$start
### 1. Total altered Length (Mb)
tot_altered_len <- sum(intersection_df$len) / 1e6
### 2. Altered Ratio
altered_ratio <- tot_altered_len / as.numeric(params$exome_length)
### 3. Total Num. Segments
num_segs <- nrow(segs_df)
### 4. Average Altered Length (kb)
ave_alt_len <- tot_altered_len * 1e3 / num_segs

cnv_burden <- data.frame(tot_altered_len,
                         altered_ratio,
                         num_segs,
                         ave_alt_len)

######### Report metrics
cnv_burden2 <- cbind(round(tot_altered_len, 2), 
                     round(altered_ratio * 100, 2), 
                     as.character(num_segs), 
                     round(ave_alt_len, 2))
colnames(cnv_burden2) <- c("Total Altered Length (Mbp)", 
                           "Altered Proportion of Exome (%)", 
                           "Total Number of Alterations", 
                           "Average Length of Alteraions (kbp)")
kable_styling(kable(t(cnv_burden2),
                    format = "latex", 
                    booktabs = TRUE, 
                    col.names = NULL,
                    align = "c"))
```

```{r SCNA_burden_save}
SCNA_burden_tbl_path <- file.path(params$script_dir, 
                                  paste0(params$type, 
                                         "_SCNA_burden_table.RDS"))
if (!file.exists(SCNA_burden_tbl_path)) {
  scna_burden_df <- data.frame()
} else {
  scna_burden_df <- readRDS(SCNA_burden_tbl_path)
}

## add to SCNA burden table if not present
if(!params$ID %in% scna_burden_df$ID) {
  
  current_df <- data.frame(ID = params$ID, 
                           primary = params$primary_tm)
  current_df <- cbind(current_df, cnv_burden)
  scna_burden_df <- rbind(scna_burden_df, current_df)
  saveRDS(scna_burden_df, SCNA_burden_tbl_path)
}
```

`r if(params$type == "glioma") {"### IV. b. Important SCNAs in Glioma"} else {"### IV. b. Important SCNAs in Medulloblastoma"}`

`r if(params$type == "glioma") {"This is the intersection of SCNAs in this case with a list of SCNAs manually curated because of their importance in gliomas."} else {"This is the intersection of SCNAs in this case with a list of SCNAs manually curated because of their importance in medulloblastomas."}`

```{r imp_cna}
if(file.exists("./NOTATES/SCNA/important_CNVs.csv")) {
  imp_cna <- read.csv("./NOTATES/SCNA/important_CNVs.csv")
  kable_styling(kable(imp_cna,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
} else {
  cat("No variation to report.")
}
```

### IV. c. SCNAs in Cancer Gene Census genes

```{r cgc_cna}
if(file.exists("./NOTATES/SCNA/CGC_CNVs.csv")) {
  cgc_cna <- read.csv("./NOTATES/SCNA/CGC_CNVs.csv")
  kable_styling(kable(cgc_cna,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
} else {
  cat("No variation to report.")
}
```

### IV. d. Broad SCNAs

```{r brod_cna}
if(file.exists("./NOTATES/SCNA/broad.csv")) {
  broad_cna <- read.csv("./NOTATES/SCNA/broad.csv")
  colnames(broad_cna)[1] <- "chromosome:start-end"
  kable_styling(kable(broad_cna,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
} else {
  cat("No variation to report.")
}
```

\newpage

### IV. e. SCNA by Chromosome

\begin{center}
  \makebox[\textwidth]{\includegraphics[width=\paperwidth]{./ExomeCNV/{CNV.cnv}.png}}
\end{center}

\newpage

# V. Loss of Heterozygosity (LOH) Events

Only LOH events for which absolute difference of B-allele frequencies ($|BAF_{Tumor}-BAF_{Normal}|$) is larger than 0.4 are reported.

### V. a. LOH Overview

```{r loh_regions}
loh <- read.csv("./NOTATES/LOH/LOH.csv")
loh$Position <- paste0(loh$Chr, ":", loh$Start, "-", loh$End)
loh <- loh[,c("Genes", "Position", "N_BAF", "T_BAF", "Absolute_Diff")]

kable_styling(kable(loh,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
```

### V. b. LOH + somatic SNV/Indel

```{r loh_dhit}
if(file.exists("./NOTATES/LOH/d_hit_loh.csv"))
{
  loh_dhit <- read.csv("./NOTATES/LOH/d_hit_loh.csv")
  kable_styling(kable(loh_dhit,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
}else
{
  cat("No event to report.")
}
```


### V. c. LOH Events in CGC genes 

```{r loh_cgc}
if(file.exists("./NOTATES/LOH/CGC_loh.csv"))
{
  loh_cgc <- read.csv("./NOTATES/LOH/CGC_loh.csv")
  kable_styling(kable(loh_cgc,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
} else {
  cat("No event to report.")
}
```

\newpage

# VI. Genes with Double Hit

In this section, the list of genes with somatic SNV/indel(s) as well as SCNA and/or LOH events are reported. A double hit strongly suggests a relevant tumor-suppressor gene.

```{r dhit_tbl}
if(file.exists("./NOTATES/double_hit.csv"))
{
  dhit_tbl <- read.csv("./NOTATES/double_hit.csv")
  kable_styling(kable(dhit_tbl,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
} else {
  cat("No event to report.")
}
```


# VII. Tumor Heterogeneity Analysis 
```{r n3_or_n2}
if(!file.exists("./THetA/output/CNV.BEST.results")){
   n3_chosen <- F
}else{
  theta_out <- read.delim("./THetA/output/CNV.BEST.results")
  n3_chosen <- ifelse(length(unlist(strsplit(theta_out[1,"mu"], ","))) == 2, F, T)
}
```

THetA (Tumor Heterogeneity Analysis) is an algorithm that estimates the tumor purity and clonal/sublconal copy number aberrations directly from high-throughput DNA sequencing data.

\begin{center} 
\includegraphics[width=500pt]{`r ifelse(n3_chosen,"./THetA/output/CNV.n3.graph.pdf","./THetA/output/CNV.n2.graph.pdf")`}
\end{center}

\newpage

# VIII. Mutational Signatures

\begin{center} 
\includegraphics[width=500pt]{mut_signatures.pdf}
\end{center}

\begin{center} 
\includegraphics[width=200pt]{mut_signatures_pie.pdf}
\end{center}

```{r mut_sigs}
mut_sigs <- read.csv("./mut_signatures.csv", row.names = 1)
k_tbl <- kable(mut_sigs, format = "latex", booktabs = TRUE)
k_tbl <- column_spec(k_tbl, 3, width = "30em")
k_tbl <- column_spec(k_tbl, 4, width = "30em")
k_tbl <- column_spec(k_tbl, 5, width = "30em")
kable_styling(k_tbl, latex_options = "scale_down")
```

\newpage

# IX. pathfindR - KEGG Pathway Enrichment Analysis

Pathway enrichment analysis is frequently utilized for studying the mechanisms underlying oncological processes. For this analysis KEGG pathway enrichment analysis was performed using the active-subnetwork-oriented enrichment approach of `pathfindR`. `pathfindR` leverages interaction information from a protein-protein interaction network (PIN) to identify distinct active subnetworks ^[An active subnetwork can be defined as a group of interconnected genes in a PIN that predominantly consists of significantly altered genes] and then perform enrichment analyses on these subnetworks. 



### IX. a. Enrichment Results for High-impact Somatic SNV/indels

Genes that harbor non-synonymous exonic mutations are included in the enrichment analysis.

\begin{center}
\includegraphics[width=300pt]{./pathfindR_results/HQ_mutations/{enrichment_chart}.png}
\end{center}

```{r enr_res_mut}
enr_res <- read.csv("pathfindR_results/HQ_mutations/mut_enrichment_results.csv", row.names = 1)
enr_res$Down_regulated <- NULL
colnames(enr_res)[7] <- "Somatic SNV/indel"
kable_styling(kable(enr_res,
                    format = "latex", booktabs = TRUE), 
              latex_options = "scale_down")
```


### IX. b. Enrichment Results for High-impact SCNA

Genes that harbor homozygous deletion or multi-copy amplification (3+) are included in the enrichment analysis.

\begin{center}
\includegraphics[width=300pt]{./pathfindR_results/HQ_SCNA/{enrichment_chart}.png}
\end{center}

```{r enr_res_SCNA}
enr_res <- read.csv("pathfindR_results/HQ_SCNA/SCNA_enrichment_results.csv", row.names = 1)
colnames(enr_res)[7:8] <- c("Amplification", "Deletion")
kable_styling(kable(enr_res,
                    format = "latex", booktabs = TRUE), 
              latex_options = "scale_down")
```
