---
output: pdf_document
params:
  ID: ""
  script_dir: ""
  exome_length: ""
  type: ""
header-includes:
- \usepackage{titling}
- \pretitle{\begin{center}\LARGE\includegraphics[width=160pt]{`r paste0(params$script_dir, "/NOT_logo.png")`}\\[\bigskipamount]}
- \posttitle{\end{center}}
- \usepackage{pdflscape}
- \newcommand{\blandscape}{\begin{landscape}}
- \newcommand{\elandscape}{\end{landscape}}
---

---
title: "\n\n\n\nNeuroOncology Technologies (NOT)"
author: "Whole Exome Sequencing Report"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
options(stringsAsFactors = FALSE)
library(knitr)
library(pander)
```

\pagenumbering{gobble} 
\vspace{3cm}

***

>>> `r paste("**NOT ID: **", params$ID)` \
**Tumor sample:** Fresh tumor tissue, snap-frozen in liquid nitrogen \
**Normal sample:** Peripheral venous blood \
**DNA extraction method:** QIAGEN DNeasy Blood & Tissue kit \
**Report date:** `r format(Sys.time(), '%d %B, %Y')`

***


\raggedright
\clearpage
\pagenumbering{arabic}


# I. Quality Metrics
```{r QC, echo = F}
QC_df <- read.csv("./QC_table.csv", row.names = 1)
kable(QC_df, align = "c")
```

>>> \* PF is defined as passing Illumina's filter \
\*\* The percentage of PF reads that aligned to the reference sequence \
\*\*\* The percentage of reads whose mate pair was also aligned to the reference


\newpage
\blandscape

# II. Germline Alterations

In this section, the germline SNV/indels in this sample is reported. Only the alterations with ClinVar clinical significances containing "Pathogenic" designations are reported`r if(params$type == "glioma") {" (except for the subsection II. f. Common Variants)"}`.

### II. a. ACMG Incidental Findings
```{r, ACMG, echo = F, comment=NA}
if(file.exists("./NOTATES/Germline/ACMG.csv"))
{
  ACMG_df <- read.csv("./NOTATES/Germline/ACMG.csv")
  rownames(ACMG_df) <- ACMG_df$ID
  ACMG_df <- ACMG_df[,-2]
  pander(ACMG_df, align = "c",table.caption.prefix ="")
}else
{
  cat("No variation to report.")
}
```

### II. b. Variations in Cancer Gene Census genes
```{r, CGC, echo = F, comment=NA}
if(file.exists("./NOTATES/Germline/CGC.csv"))
{
  gCGC_df <- read.csv("./NOTATES/Germline/CGC.csv")
  pander(gCGC_df, align = "c",table.caption.prefix ="")
}else
{
  cat("No variation to report.")
}
```

### II. c. Variations in Cancer Predisposition Genes
```{r, CPG, echo = F, comment=NA}
if(file.exists("./NOTATES/Germline/CPG.csv"))
{
  CPG_df <- read.csv("./NOTATES/Germline/CPG.csv")
  pander(CPG_df, align = "c",table.caption.prefix ="")
}else
{
  cat("No variation to report.")
}
```

### II. d. Variations in Fanconi Anemia Pathway genes
```{r, FAP, echo = F, comment=NA}
if(file.exists("./NOTATES/Germline/FAP.csv"))
{
  FAP_df <- read.csv("./NOTATES/Germline/FAP.csv")
  pander(FAP_df, align = "c",table.caption.prefix ="")
}else
{
  cat("No variation to report.")
}
```

### II. e. Other important germline variations
```{r, OTH, echo = F, comment=NA}
if(file.exists("./NOTATES/Germline/OTHER.csv"))
{
  OTH_df <- read.csv("./NOTATES/Germline/OTHER.csv")
  pander(OTH_df, align = "c",table.caption.prefix ="")
}else
{
  cat("No variation to report.")
}
```

`r if(params$type == "glioma") {"### II. f. Common Variants"}`

```{r, common, echo = F, comment=NA}
if (params$type == "glioma") {
  if(file.exists("./NOTATES/Germline/common_var.csv"))
  {
    common_df <- read.csv("./NOTATES/Germline/common_var.csv")
    pander(common_df, align = "c",table.caption.prefix ="")
  }else
  {
    cat("No variation to report.")
  }
}
```

\newpage

# III. Somatic Single Nucleotide Variations(SNVs)

### III. a. Mutational Load

```{r mutload, echo=F, fig.width=12, fig.height=2.5}
somatic_SNVs <- read.delim("./Oncotator/annotated.sSNVs.tsv", comment.char="#")

# Subsetting for (MuTect's default) HQ filters
somatic_SNVs <- subset(somatic_SNVs, 
                       alt_allele_seen=="True" & 
                       short_tandem_repeat_membership == "False")
mut_num <- nrow(somatic_SNVs)
mut_load <- mut_num/as.numeric(params$exome_length)

if (params$type == "glioma") {
  mut_load_df <- read.delim(paste0(params$script_dir, "/mut_loads.txt"), sep = "\t") 
  
  if(!params$ID %in% mut_load_df$Patient) {
    somatic_SNVs <- somatic_SNVs[! somatic_SNVs$Variant_Classification %in% c("Silent", "3'UTR", "3'Flank", "5'UTR", "5'Flank", 
                                                                              "IGR", "Intron", "lincRNA", "RNA"),]
    sset <- ifelse("H3F3A" %in% somatic_SNVs$Hugo_Symbol, "H3-mut",
                   ifelse("IDH1" %in% somatic_SNVs$Hugo_Symbol | "IDH2" %in% somatic_SNVs$Hugo_Symbol, "IDH-mut", "IDH-WT"))
    
    mut_load_df <- rbind(mut_load_df, data.frame(Patient = params$ID, 
                                                 mutational_load = mut_load, 
                                                 subset = sset))
    write.table(mut_load_df, paste0(params$script_dir, "/mut_loads.txt"), sep = "\t", row.names = F, quote = F)
  }
  
  mut_load_df <- mut_load_df[order(mut_load_df$mutational_load),]
  mut_load_df$x_val <- 1:nrow(mut_load_df)
  
  suppressPackageStartupMessages(library(ggplot2))
  
  p <- ggplot(mut_load_df, aes(x=x_val, y=mutational_load, group = subset))
  p <- p + geom_point(size=2, aes(shape=subset, color=subset))
  p <- p + scale_color_manual(values=c('#56B4E9','#E69F00', 'gray60'))
  p <- p + geom_segment(aes(x = median(mut_load_df$x_val)-6, y = median(mut_load_df$mutational_load), 
                            xend = median(mut_load_df$x_val)+6, yend = median(mut_load_df$mutational_load)), colour = "red")
  p <- p + coord_flip()
  p <- p + theme(axis.title.y=element_blank(),
                 axis.text.y=element_blank(),
                 axis.ticks.y=element_blank())
  p <- p + scale_y_log10(breaks = scales::trans_breaks("log10", function(x) 10^x),
                         labels = scales::trans_format("log10", scales::math_format(10^.x)))
  p <- p + ylab("Mutational Load")
  p <- p + geom_point(data=mut_load_df[mut_load_df$Patient == params$ID, ], aes(x=x_val, y=mutational_load), colour="red", size=4, shape = 1)
  p
}
```

`r if(params$type == "glioma") {paste0("Mutational loads (Numbers of mutations per Mb) for ", nrow(mut_load_df), " patients (including the current patient, circled in red) are represented in the above scatter plot. Red vertical line segment represents the overall median mutational load value.")}`

```{r load_table, echo = F, comment=NA}
if (params$type == "glioma") {
  load_tbl <- tapply(mut_load_df$mutational_load, mut_load_df$subset, median)
  load_tbl <- as.data.frame(load_tbl)
  colnames(load_tbl) <- "Median Mutational Load"
  
  load_tbl <- rbind(load_tbl, median(mut_load_df$mutational_load))
  rownames(load_tbl)[nrow(load_tbl)] <- "Overall"
  
  pander(load_tbl, align = "c",table.caption.prefix ="")
}
```

** Mutation load (Number of mutations per Mb) for the current patient is: `r mut_load` **

### III. b. Microsatellite Instability Status

```{r variables, echo = F, comment=NA}
var_tbl <- read.csv("MSIpred/maf_feats.csv")
rownames(var_tbl) <- NULL
var_tbl <- round(var_tbl[1, -1], 2)
pander(var_tbl, align = "c", table.caption.prefix ="")
```

```{r classification, echo = F, comment=NA}
class_tbl <- read.csv("MSIpred/MSIpred_prediction.csv", row.names = 1)
rownames(class_tbl) <- NULL
pander(class_tbl, align = "c", table.caption.prefix ="")
```

`r if(params$type == "glioma") {"### III. c. Important SNVs in Glioma"} else {"### III. c. Important SNVs in Medulloblastoma"}`

`r if(params$type == "glioma") {"This is the intersection of SNVs in this case with a list of SNVs manually curated because of their importance in gliomas."} else {"This is the intersection of SNVs in this case with a list of SNVs manually curated because of their importance in medulloblastomas."}`


```{r imp_snv, echo = F, comment=NA}
if(file.exists("./NOTATES/Somatic_SNV/important_SNVs.csv"))
{
  imp_snv <- read.csv("./NOTATES/Somatic_SNV/important_SNVs.csv", colClasses = c("character"))
  imp_snv <- imp_snv[,setdiff(colnames(imp_snv),c("COSMIC_total_alterations_in_gene","DNA_repair","selected_KEGG"))]
  colnames(imp_snv) <- c("Gene", "Effect", "Protein", "Genome","TumorF","Genotype", "COS_n_ov",
                        "UniProt_Region", "SIFT","Irino", "TMZ")
  
  pander(imp_snv, align = "c",table.caption.prefix ="")
}else
{
  cat("No variation to report.")
}
```

### III. d. CGC gene + COSMIC hotspot

These are the SNVs in this case in a CGC gene, positioned at a site with at least one COSMIC overlapping mutation.

```{r, hotspot, echo = F, comment=NA}
if(file.exists("./NOTATES/Somatic_SNV/COSMIC_hotspot.csv"))
{
  hot_snv <- read.csv("./NOTATES/Somatic_SNV/COSMIC_hotspot.csv", colClasses = c("character"))
  hot_snv <- hot_snv[,setdiff(colnames(hot_snv),c("COSMIC_total_alterations_in_gene","DNA_repair","selected_KEGG"))]
  colnames(hot_snv) <- c("Gene", "Effect", "Protein", "Genome","TumorF","Genotype", "COS_n_ov",
                        "UniProt_Region", "SIFT", "Irino", "TMZ")

  pander(hot_snv, align = "c",table.caption.prefix ="")
}else
{
  cat("No variation to report.")
}
```

### III. e. CGC gene only

These are the SNVs in a CGC gene, not positioned at a site with any COSMIC overlapping mutations.

```{r, cgc_snv, echo = F, comment=NA}
if(file.exists("./NOTATES/Somatic_SNV/CGC_genes.csv"))
{
  cgc_snv <- read.csv("./NOTATES/Somatic_SNV/CGC_genes.csv", colClasses = c("character"))
  cgc_snv <- cgc_snv[,setdiff(colnames(cgc_snv),c("COSMIC_total_alterations_in_gene","DNA_repair","selected_KEGG"))]
  colnames(cgc_snv) <- c("Gene", "Effect", "Protein", "Genome","TumorF","Genotype", "COS_n_ov",
                        "UniProt_Region", "SIFT", "Irino", "TMZ")
  pander(cgc_snv, align = "c",table.caption.prefix ="")
}else
{
  cat("No variation to report.")
}
```

### III. f. Other possibly important SNVs

SNVs in DNA damage repair genes:

```{r, DDR, echo = F, comment=NA}
if(file.exists("./NOTATES/Somatic_SNV/DDR_related.csv"))
{
  ddr_snv <- read.csv("./NOTATES/Somatic_SNV/DDR_related.csv", colClasses = c("character"))

  ddr_snv <- ddr_snv[,setdiff(colnames(ddr_snv),c("COSMIC_total_alterations_in_gene","selected_KEGG"))]
  colnames(ddr_snv) <- c("Gene", "Effect", "Protein", "Genome","TumorF","Genotype", "COS_n_ov",
                        "UniProt_Region", "SIFT", "DNA_repair", "Irino", "TMZ")

  pander(ddr_snv, align = "c",table.caption.prefix ="")
}else
{
  cat("No variation to report.")
}
```

SNVs in important KEGG Pathway genes:

```{r, kegg, echo = F, comment=NA}
if(file.exists("./NOTATES/Somatic_SNV/KEGG_selected.csv"))
{
  kegg_snv <- read.csv("./NOTATES/Somatic_SNV/KEGG_selected.csv", colClasses = c("character"))

  colnames(kegg_snv) <- c("Gene", "Effect", "Protein", "Genome","TumorF","Genotype", "COS_n_ov",
                          "COS_tot", "UniProt_Region", "SIFT", "DNA_repair", "Irino", "TMZ", "KEGG")

  pander(kegg_snv, align = "c",table.caption.prefix ="")
}else
{
  cat("No variation to report.")
}
```

SNVs in important Irinotecan response genes:

```{r, irino, echo = F, comment=NA}
if(file.exists("./NOTATES/Somatic_SNV/irinotecan.csv"))
{
  irino_snv <- read.csv("./NOTATES/Somatic_SNV/irinotecan.csv", colClasses = c("character"))

  colnames(irino_snv) <- c("Gene", "Effect", "Protein", "Genome","TumorF","Genotype", "COS_n_ov",
                          "COS_tot", "UniProt_Region", "SIFT", "DNA_repair", "Irino", "TMZ", "KEGG")
  
  pander(irino_snv, align = "c", table.caption.prefix ="")
}else
{
  cat("No variation to report.")
}
```

SNVs in important Temozolomide resistance genes:

```{r, tmz, echo = F, comment=NA}
if(file.exists("./NOTATES/Somatic_SNV/tmz_res.csv"))
{
  tmz_snv <- read.csv("./NOTATES/Somatic_SNV/tmz_res.csv", colClasses = c("character"))
  
  colnames(tmz_snv) <- c("Gene", "Effect", "Protein", "Genome","TumorF","Genotype", "COS_n_ov",
                          "COS_tot", "UniProt_Region", "SIFT", "DNA_repair", "Irino", "TMZ", "KEGG")

  pander(tmz_snv, align = "c",table.caption.prefix ="")
}else
{
  cat("No variation to report.")
}
```

Other coding mutations and non-coding mutations are provided as attachments.

\newpage

# IV. Somatic Copy Number Alterations (SCNAs)

Only SCNAs with a $log_2(Tumor/Normal)$ ratio $\le$ -0.25 or $\ge$ 0.2 are reported in the tables.

`r if(params$type == "glioma") {"### IV. a. Important SCNAs in Glioma"} else {"### IV. a. Important SCNAs in Medulloblastoma"}`

`r if(params$type == "glioma") {"This is the intersection of SCNAs in this case with a list of SCNAs manually curated because of their importance in gliomas."} else {"This is the intersection of SCNAs in this case with a list of SCNAs manually curated because of their importance in medulloblastomas."}`

```{r, imp_cna, echo = F, comment=NA}
if(file.exists("./NOTATES/SCNA/important_CNVs.csv"))
{
  imp_cna <- read.csv("./NOTATES/SCNA/important_CNVs.csv")
  pander(imp_cna, align = "c",table.caption.prefix ="")
}else
{
  cat("No variation to report.")
}
```

### IV. b. SCNAs in Cancer Gene Census genes

```{r, cgc_cna, echo = F, comment=NA}
if(file.exists("./NOTATES/SCNA/CGC_CNVs.csv"))
{
  cgc_cna <- read.csv("./NOTATES/SCNA/CGC_CNVs.csv")
  pander(cgc_cna[,-c(2,3)], align = "c",table.caption.prefix ="")
}else
{
  cat("No variation to report.")
}
```

### IV. c. Broad SCNAs

```{r, brod_cna, echo = F, comment=NA}
if(file.exists("./NOTATES/SCNA/broad.csv"))
{
  broad_cna <- read.csv("./NOTATES/SCNA/broad.csv")
  pander(broad_cna, align = "c",table.caption.prefix ="")
}else
{
  cat("No variation to report.")
}
```

### IV. d. SCNA by Chromosome Plots

\begin{center} 
\includegraphics[width=600pt]{./ExomeCNV/CNV.cnv.png}
\end{center}

\newpage

# V. Loss of Heterozygosity (LOH) Events

Only LOH events for which absolute difference of B-allele frequencies ($|BAF_{Tumor}-BAF_{Normal}|$) is larger than 0.4 are reported.

### V. a. LOH Overview

```{r, loh_regions, echo = F, comment=NA}
loh <- read.csv("./NOTATES/LOH/LOH.csv")
loh$Position <- paste0(loh$Chr, ":", loh$Start, "-", loh$End)
loh <- loh[,c("Genes", "Position", "N_BAF", "T_BAF", "Absolute_Diff")]
pander(loh, align = "c",table.caption.prefix ="")
```

### V. b. LOH Events in CGC genes 

```{r, loh_cgc, echo = F, comment=NA}
if(file.exists("./NOTATES/LOH/CGC_loh.csv"))
{
  loh_cgc <- read.csv("./NOTATES/LOH/CGC_loh.csv")
  pander(loh_cgc, align = "c",table.caption.prefix ="")
}else
{
  cat("No event to report.")
}
```

\newpage

# VI. Tumor Heterogeneity Analysis 
```{r n3_or_n2, echo=F}
if(!file.exists("./THetA/output/CNV.BEST.results")){
   n3_chosen <- F
}else{
  theta_out <- read.delim("./THetA/output/CNV.BEST.results")
  n3_chosen <- ifelse(length(unlist(strsplit(theta_out[1,"mu"], ","))) == 2, F, T)
}
```

THetA (Tumor Heterogeneity Analysis) is an algorithm that estimates the tumor purity and clonal/sublconal copy number aberrations directly from high-throughput DNA sequencing data.

\begin{center} 
\includegraphics[width=500pt]{`r ifelse(n3_chosen,"./THetA/output/CNV.n3.graph.pdf","./THetA/output/CNV.n2.graph.pdf")`}
\end{center}

\elandscape
\newpage

# VII. Mutational Signatures

\begin{center} 
\includegraphics[width=500pt]{mut_signatures.pdf}
\end{center}

\begin{center} 
\includegraphics[width=400pt]{mut_signatures_pie.pdf}
\end{center}

\newpage
\blandscape

```{r, mut_sigs, echo = F, comment=NA}
mut_sigs <- read.csv("./mut_signatures.csv", row.names = 1)
pander(mut_sigs, align = "c",table.caption.prefix ="")
```

\newpage

# VIII. KEGG Pathway Enrichment Analysis

Genes that harbor non-synonymous exonic mutations and/or homozygous deletion or multi-copy amplification (3+) are included in the enrichment analysis.

### VIII. a. KEGG - Pathways in Cancer
\begin{center} 
\includegraphics[width=500pt]{./pathways/hsa05200.CANCER.multi.png}
\end{center}

The right half of a gene indicates copy number ratio (scaled between -1 and 1) while the left half indicates somatic mutation status (1 if mutant).

### VIII. b. KEGG - Pathway Enrichment Results

```{r, enr_res, echo = F, comment=NA}
enr_res <- read.csv("pathways/enrichment_results.csv", row.names = 1)
pander(enr_res, align = "c",table.caption.prefix ="")
```

\elandscape
