---
output: pdf_document
params:
  ID: ""
  script_dir: ""
  exome_length: ""
  exome_bed: ""
  type: ""
  primary_tm: ""
  tumor_sample: ""
header-includes:
- \usepackage{titling}
- \pretitle{\begin{center}\LARGE\includegraphics[width=160pt]{`r file.path(params$script_dir, "NOT_logo.png")`}\\[\bigskipamount]}
- \posttitle{\end{center}}
- \usepackage{float}
- \usepackage{booktabs}
- \usepackage{tabu}
- \usepackage{longtable}
title: \ \ NeuroOncology Technologies (NOT)
subtitle: "Whole Exome Sequencing Report"
---

```{r setup, include=FALSE}
options(kableExtra.latex.load_packages = FALSE)
knitr::opts_chunk$set(echo = FALSE, comment = NA)
options(stringsAsFactors = FALSE, Encoding="UTF-8")
suppressPackageStartupMessages(library(knitr))
suppressPackageStartupMessages(library(kableExtra))
suppressPackageStartupMessages(library(GenomicRanges))
```

\pagenumbering{gobble} 
\vspace{2cm}

***

>>> `r paste("**NOT ID: **", params$ID)` \
**Treatment status:** `r ifelse(as.logical(params$primary_tm), "Primary (untreated) tumor", "Recurrent (previously treated) tumor")` \
**Tumor sample:** `r ifelse(params$tumor_sample == "FFPE", "Formalin-Fixed Paraffin-Embedded (FFPE) tissue specimen", "Fresh tumor tissue, snap-frozen in liquid nitrogen")` \
**Normal sample:** Peripheral venous blood \
**DNA extraction method:** QIAGEN DNeasy Blood & Tissue kit \
**Report date:** `r format(Sys.time(), '%d %B, %Y')`

***


\raggedright
\clearpage
\pagenumbering{arabic}


# I. Quality Metrics

## Ia. Summary Table of Quality Metrics

```{r QC}
QC_df <- read.csv("./QC_table.csv" )
colnames(QC_df)[1] <- ""
kable_styling(kable(QC_df,
                    format = "latex", 
                    booktabs = TRUE,
                    align = c("r", "c", "c")))
```

>>> \* PF is defined as passing Illumina's filter \
    \*\* The percentage of PF reads that aligned to the reference sequence \
    \*\*\* The percentage of reads whose mate pair was also aligned to the reference

## Ib. Tumor Purity

The fraction of reads coming from cross-sample contamination, reflecting a measure of tumor purity, is calculated using the GATK - `CalculateContamination` tool. This tool calculates the fraction of reads coming from cross-sample contamination and estimates contamination based on the signal from reference reads at homozygous alternate sites. A purity/clonality estimate (reflecting normal contamination in the tumor sample) based on copy number alterations (CNAs) is presented under section VII. Tumor Heterogeneity Analysis.

```{r contamination}
contam_df <- read.delim("contamination.table")
kable_styling(kable(contam_df[, -1],
                    format = "latex",
                    booktabs = TRUE,
                    align = "c"))
```

\newpage

# II. Germline Alterations

Germline alterations are assessed from the WES sequencing of DNA extracted from blood. Alterations are called using GATK - HaplotypeCaller. Findings are filtered for germline single nucleotide variations (SNVs) and short (typically less than 20 bases-long) insertion-deletion events (indels) that:

- were not reported as "benign" or "likely benign" in ClinVar
- have population Minor Allele Frequency (MAF) < 1% (in 1000 Genomes, gnomAD, ExAC and ESP6500)
- have non-synonymous impact (one of "Frame_Shift_Del", "Frame_Shift_Ins", "Splice_Site", "Translation_Start_Site","Nonsense_Mutation", "Nonstop_Mutation", "In_Frame_Del","In_Frame_Ins", "Missense_Mutation", see [http://www.ensembl.org/Help/Glossary?id=535]( http://www.ensembl.org/Help/Glossary?id=535)))
- are not in genes that are often non-pathogenic and passengers but are frequently mutated in most of the public exome studies (named FLAGS) as collected by Shyr et al. [^1]. 

[^1]: Shyr C, Tarailo-graovac M, Gottlieb M, Lee JJ, Van karnebeek C, Wasserman WW. FLAGS, frequently mutated genes in public exomes. BMC Med Genomics. 2014;7:64.

`r if(params$type == "glioma") {" (except for the subsection \"Common Variants\")"}`. The altered gene, the position of the SNV, the reference nucleotide as well as the altered nucleotide, the mutation type, the read depth of the variation, the estimated allelic frequency and clinical significance (as reported in ClinVar) are reported. The report follows a sequential order where an alteration reported in a subsection is not reported in the following subsections.

## II. a. ACMG Incidental Findings

Filtered germline SNV/indels that affects any of the genes listed in ACMG SF V2.0[^2] are reported under this subsection.

[^2]: Kalia SS, Adelman K, Bale SJ, et al. Recommendations for reporting of secondary findings in clinical exome and genome sequencing, 2016 update (ACMG SF v2.0): a policy statement of the American College of Medical Genetics and Genomics. Genet Med. 2017;19(2):249-255.

```{r ACMG}
if(file.exists("./NOTATES/Germline/ACMG.csv")) {
  ACMG_df <- read.csv("./NOTATES/Germline/ACMG.csv")
  rownames(ACMG_df) <- ACMG_df$ID
  ACMG_df <- ACMG_df[,-2]
  
  kable_styling(kable(ACMG_df,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"),
                latex_options="scale_down")
} else {
  cat("No variation to report.")
}
```

## II. b. Variants in Cancer Gene Census Genes

Cancer Gene Census (CGC) from the Catalogue of Somatic Mutations in Cancer (COSMIC) database is a catalogue of genes which contain mutations that have been causally linked to cancer [(https://cancer.sanger.ac.uk/census)](https://cancer.sanger.ac.uk/census). This subsection filters the germline SNV/indels for genes that are reported in CGC. 

```{r CGC}
if(file.exists("./NOTATES/Germline/CGC.csv")) {
  gCGC_df <- read.csv("./NOTATES/Germline/CGC.csv")
  kable_styling(kable(gCGC_df,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"),
                latex_options="scale_down")
} else {
  cat("No variation to report.")
}
```

## II. c. Variants in Cancer Predisposition Genes

Genes in which germline mutations confer highly or moderately increased risks of cancer are called cancer predisposition genes. Filtered germline SNV/indels catalogued in the 2014 Nature publication by Rahman [^3] are reported in this subsection.


[^3]: Rahman N. Realizing the promise of cancer predisposition genes. Nature. 2014;505(7483):302-8.

```{r CPG}
if(file.exists("./NOTATES/Germline/CPG.csv")) {
  CPG_df <- read.csv("./NOTATES/Germline/CPG.csv")
  kable_styling(kable(CPG_df,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"),
                latex_options="scale_down")
} else {
  cat("No variation to report.")
}
```


## II. d. Variants in DNA Damage Repair Genes

In this subsection, filtered germline variants in Human DNA Repair Genes [^4] are presented.

[^4]: Mdanderson.org. 2020. Human DNA Repair Genes. [online] Available at: <https://www.mdanderson.org/documents/Labs/Wood-Laboratory/human-dna-repair-genes.html>.

```{r DDR_germ}
if(file.exists("./NOTATES/Germline/DDR.csv")) {
  DDR_df <- read.csv("./NOTATES/Germline/DDR.csv")
  kable_styling(kable(DDR_df,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"),
                latex_options="scale_down")
} else {
  cat("No variation to report.")
}
```

`r if(params$type == "glioma") {"## II. e. Common Variants"}`

`r if(params$type == "glioma") {"This subsection is distinct from the previous germline alterations subsections. Here, germline alterations are filtered for “single nucleotide polymorphisms (SNPs)” which were previously shown in genome wide association studies (GWAS) to have an association with gliomas and were listed in the GWAS catalog [(https://www.ebi.ac.uk/gwas/efotraits/EFO_0005543)](https://www.ebi.ac.uk/gwas/efotraits/EFO_0005543)"}`

```{r common}
if (params$type == "glioma") {
  if(file.exists("./NOTATES/Germline/common_var.csv")) {
    common_df <- read.csv("./NOTATES/Germline/common_var.csv")
    kable_styling(kable(common_df,
                        format = "latex", 
                        booktabs = TRUE,
                        align = "c"))
  } else {
    cat("No variation to report.")
  }
}
```

\newpage

# III. Somatic Single Nucleotide Variations (SNVs) and Small Insertion/Deletions (Indels)

Somatic mutations were called from the tumor-blood pair in the current analysis using GATK - MuTect2 with a cutoff of 5% variant allele frequency (VAF). Only non-synonymous variants (with variant classifications of High/Moderate variant consequences: "Frame_Shift_Del", "Frame_Shift_Ins", "Splice_Site", "Translation_Start_Site","Nonsense_Mutation", "Nonstop_Mutation", "In_Frame_Del","In_Frame_Ins", "Missense_Mutation", see [http://www.ensembl.org/Help/Glossary?id=535]( http://www.ensembl.org/Help/Glossary?id=535)) were used for reporting. Genes that are often non-pathogenic and passengers but are frequently mutated in most of the public exome studies (named FLAGS) as collected by Shyr et al. were excluded from the report. 

## III. a. Tumor Mutational Burden (TMB)

TMB is defined as the number of somatic mutations in the coding region (more accurately the exome capture kit's intervals) per megabase, including SNVs and indels.

This calculation was performed using the following filtering approach:

1. keeping variants with mutated allele frequency greater than 5%
2. keeping variants with a sequence depth greater than 20X in the tumor sample greater and 10X in the normal sample

```{r TMB, fig.width=10, fig.height=3}
somatic_SNVs <- read.delim("./Funcotator/annotated_somatic.maf", comment.char="#")

somatic_SNVs$tot_depth_tumor <- somatic_SNVs$t_alt_count + somatic_SNVs$t_ref_count
somatic_SNVs$tot_depth_normal <- somatic_SNVs$n_alt_count + somatic_SNVs$n_ref_count

# Filter for depth greater than 20X in tumor samples greater or 10X in normal samples
somatic_SNVs <- somatic_SNVs[somatic_SNVs$tot_depth_tumor > 20 & somatic_SNVs$tot_depth_normal > 10, ]

# subset for tumor_f > 0.05
somatic_SNVs <- somatic_SNVs[somatic_SNVs$tumor_f > 0.05, ]

## Calculate TMB
mut_num <- nrow(somatic_SNVs)
current_TMB <- mut_num / as.numeric(params$exome_length)

TMB_tbl_path <- file.path(params$script_dir, 
                          paste0(params$type, "_TMB_table.RDS"))
if (!file.exists(TMB_tbl_path)) {
  TMB_df <- data.frame()
} else {
  TMB_df <- readRDS(TMB_tbl_path)
}

## add to TMB table if not present
if(!params$ID %in% TMB_df$ID) {
  high_conseq <- c("Frame_Shift_Del", "Frame_Shift_Ins", "Splice_Site", 
                   "Translation_Start_Site","Nonsense_Mutation", 
                   "Nonstop_Mutation", "In_Frame_Del","In_Frame_Ins", 
                   "Missense_Mutation")
  somatic_SNVs <- somatic_SNVs[somatic_SNVs$Variant_Classification %in% high_conseq,]
  current_sset <- ifelse("H3F3A" %in% somatic_SNVs$Hugo_Symbol, "H3-mut",
                         ifelse("IDH1" %in% somatic_SNVs$Hugo_Symbol | "IDH2" %in% somatic_SNVs$Hugo_Symbol, "IDH-mut", "IDH-WT"))
  
  TMB_df <- rbind(TMB_df, 
                  data.frame(ID = params$ID, 
                             TMB = current_TMB, 
                             subset = current_sset,
                             primary = params$primary_tm))
  saveRDS(TMB_df, TMB_tbl_path)
}

## Scatter plots of TMB over all analyses
# rank TMB vals
TMB_df <- TMB_df[order(TMB_df$TMB),]
TMB_df$x_val <- 1:nrow(TMB_df)

# prep for facets
TMB_df$for_facet <- ifelse(TMB_df$primary, "Primary", "Primary\n+\nRecurrent")
pr_tms <- TMB_df[TMB_df$for_facet == "Primary", ]
pr_tms$for_facet <- "Primary\n+\nRecurrent"
TMB_df_plot <- as.data.frame(rbind(TMB_df, pr_tms))

# calculate height of median line segment
x_shift <- max(nrow(TMB_df_plot) / 7, 3)

suppressPackageStartupMessages(library(ggplot2))
p <- ggplot(TMB_df_plot, aes(x=x_val, y=TMB, group = subset))
p <- p + geom_point(size=2, aes(shape=subset, color=subset))
p <- p + scale_color_manual(values=c('#56B4E9','#E69F00', 'gray60'))
p <- p + geom_segment(aes(x = median(x_val) - x_shift, 
                          y = median(TMB), 
                          xend = median(x_val) + x_shift, 
                          yend = median(TMB)), 
                      colour = "red")
p <- p + coord_flip()
p <- p + theme(axis.title.y=element_blank(),
               axis.text.y=element_blank(),
               axis.ticks.y=element_blank())
p <- p + scale_y_log10(breaks = scales::trans_breaks("log10", function(x) 10^x),
                       labels = scales::trans_format("log10", scales::math_format(10^.x)))
p <- p + ylab("Tumor Mutational Burden")
p <- p + geom_point(data=TMB_df_plot[TMB_df_plot$ID == params$ID, ], 
                    aes(x=x_val, y=TMB), colour="red", size=4, shape = 1)
p <- p + facet_grid(rows = vars(for_facet))
p
```

`r if(params$type == "glioma") {paste0("TMBs (mutations/Mb) of all tumors reported until now, including the current tumor, circled in red are presented in the above scatter plots. The plots separately displayed for (a) primary (untreated tumors only, top panel, n =", nrow(pr_tms), ") and (b) primary + recurrent tumors (untreated tumors and tumors previously undergone treatment, bottom panel, n=", nrow(TMB_df), "). The red vertical lines indicate median TMB for each panel. The table below displays the median TMBs per glioma molecular subsets over all previous analyses for primary and primary+recurrent tumors separately.")}`

```{r load_table}
if (params$type == "glioma") {
  ss_TMB_tbl_ALL <- tapply(TMB_df$TMB, TMB_df$subset, median)
  ss_TMB_tbl_ALL <- as.data.frame(ss_TMB_tbl_ALL)
  ss_TMB_tbl_ALL <- rbind(ss_TMB_tbl_ALL, 
                          median(TMB_df$TMB))
  rownames(ss_TMB_tbl_ALL)[nrow(ss_TMB_tbl_ALL)] <- "Overall"
  colnames(ss_TMB_tbl_ALL) <- "Primary\n+\nRecurrent"
  
  tmp <- TMB_df[TMB_df$primary, ]
  ss_TMB_tbl_prim <- tapply(tmp$TMB, tmp$subset, median)
  ss_TMB_tbl_prim <- as.data.frame(ss_TMB_tbl_prim)
  ss_TMB_tbl_prim <- rbind(ss_TMB_tbl_prim, 
                           median(tmp$TMB))
  rownames(ss_TMB_tbl_prim)[nrow(ss_TMB_tbl_prim)] <- "Overall"
  colnames(ss_TMB_tbl_prim) <- "Primary"
  
  ss_TMB_tbl <- cbind(ss_TMB_tbl_ALL, ss_TMB_tbl_prim)
  
  kable_styling(kable(ss_TMB_tbl,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
}
```

> TMB for the current tumor sample is: `r round(current_TMB, 2)` mutations/Mb (`r ifelse(current_TMB > 20, "High \\(>", "Normal \\(<=")` 20 mutations/Mb))

## III. b. Microsatellite Instability Status

The microsatellite instability (MSI) status was predicted using the tool MSIpred [^5]. MSIpred computes 22 somatic mutational features (predictors of MSI) and using these features, it predicts tumor MSI status as microsatellite instability high (MSI-H) or microsatellite stable (MSS) using a machine learning method, support vector machine (SVM). 

Additionally, polymerase-epsilon (POLE) deficiency is predicted based on the presence of both of:

1. total count of somatic SNVs / Mb > 60 AND
2. count of somatic indels in single sequence repeats / Mb < 0.18

[^5]: Wang C, Liang C. MSIpred: a python package for tumor microsatellite instability classification from tumor mutation annotation data using a support vector machine. Sci Rep. 2018;8(1):17546.


```{r classification}
class_tbl <- read.csv("MSIpred/MSIpred_prediction.csv", row.names = 1)
rownames(class_tbl) <- NULL
kable_styling(kable(class_tbl,
                    format = "latex", 
                    booktabs = TRUE,
                    align = "c"))
```

`r if(params$type == "glioma") {"## III. c. Variants in Established Glioma Genes"} else {"## III. c. Variants in Established Medulloblastoma Genes"}`

`r if(params$type == "glioma") {"This subsection contains somatic SNV/indels in genes that have been reported in the TCGA pan-glioma study of Ceccarelli et al. [^6] which analyzed 1122 WHO grade II-III and IV diffuse-gliomas."} else {"This is the intersection of somaticSNV/indels in this case with a list of genes manually curated because of their importance in medulloblastomas."}`

`r if(params$type == "glioma") {"[^6]: Ceccarelli M, Barthel FP, Malta TM, et al. Molecular Profiling Reveals Biologically Discrete Subsets and Pathways of Progression in Diffuse Glioma. Cell. 2016;164(3):550-63."}`


```{r imp_som_var}
if(file.exists("./NOTATES/Somatic_SNV/important_SNVs.csv")) {
  imp_som_var <- read.csv("./NOTATES/Somatic_SNV/important_SNVs.csv", colClasses = c("character"))
  imp_som_var <- imp_som_var[, setdiff(colnames(imp_som_var), 
                               c("COSMIC_total_alterations_in_gene", "DNA_repair", "selected_KEGG"))]
  colnames(imp_som_var) <- c("Gene", "Variant Classification", "Protein", "Genomic Change", "VAF", "COS_n_ov",
                         "UniProt_Region", "Irino", "TMZ")
  
  k_tbl <- kable(imp_som_var, format = "latex", booktabs = TRUE)
  k_tbl <- column_spec(k_tbl, which(colnames(imp_som_var) == "UniProt_Region"), width = "10em")
  kable_styling(k_tbl, latex_options = "scale_down")
} else {
  cat("No variation to report.")
}
```

## III. d. Hotspot Variants in Cancer Gene Census Genes

This subsection presents somatic SNV/indels which fulfill the following criteria:

1. The gene harboring the variants is listed in the Catalogue of Somatic Mutations in Cancer (COSMIC) - CGC, which is a catalogue of genes which contain mutations that have been causally linked to cancer [(https://cancer.sanger.ac.uk/census)](https://cancer.sanger.ac.uk/census). 
2. The variant has a mutation record in COSMIC at the same genomic position (as a recurrent mutation increases the likelihood of an actionable “gain-of function” mutation).

```{r hotspot}
if(file.exists("./NOTATES/Somatic_SNV/COSMIC_hotspot.csv")) {
  hot_som <- read.csv("./NOTATES/Somatic_SNV/COSMIC_hotspot.csv", colClasses = c("character"))
  hot_som <- hot_som[, setdiff(colnames(hot_som), 
                               c("COSMIC_total_alterations_in_gene", "DNA_repair", "selected_KEGG"))]
  colnames(hot_som) <- c("Gene", "Variant Classification", "Protein", "Genomic Change", "VAF", "COS_n_ov",
                         "UniProt_Region","Irino", "TMZ")
  k_tbl <- kable(hot_som, format = "latex", booktabs = TRUE)
  k_tbl <- column_spec(k_tbl, which(colnames(hot_som) == "UniProt_Region"), width = "10em")
  kable_styling(k_tbl, latex_options = "scale_down")
} else {
  cat("No variation to report.")
}
```

## III. e. Other Variants in Cancer Gene Census Genes

This subsection lists somatic variants where the gene harboring the variants is listed in the COSMIC-CGC, which is a catalog of genes which contain mutations that have been causally linked to cancer [(https://cancer.sanger.ac.uk/census)](https://cancer.sanger.ac.uk/census).

```{r cgc_som}
if(file.exists("./NOTATES/Somatic_SNV/CGC_genes.csv")) {
  cgc_som <- read.csv("./NOTATES/Somatic_SNV/CGC_genes.csv", colClasses = c("character"))
  cgc_som <- cgc_som[, setdiff(colnames(cgc_som), 
                               c("COSMIC_total_alterations_in_gene", "DNA_repair", "selected_KEGG"))]
  colnames(cgc_som) <- c("Gene", "Variant Classification", "Protein", "Genomic Change","VAF", "COS_n_ov",
                         "UniProt_Region", "Irino", "TMZ")
  k_tbl <- kable(cgc_som, format = "latex", booktabs = TRUE)
  k_tbl <- column_spec(k_tbl, which(colnames(cgc_som) == "UniProt_Region"), width = "10em")
  kable_styling(k_tbl, latex_options = "scale_down")
} else
{
  cat("No variation to report.")
}
```

## III. f. Other Possibly Important Somatic SNV/indels

Somatic SNV/indels in DNA damage repair genes that were presented on [https://www.mdanderson.org/documents/Labs/Wood-Laboratory/human-dna-repair-genes.html](https://www.mdanderson.org/documents/Labs/Wood-Laboratory/human-dna-repair-genes.html):

```{r DDR}
if(file.exists("./NOTATES/Somatic_SNV/DDR_related.csv")) {
  ddr_som_var <- read.csv("./NOTATES/Somatic_SNV/DDR_related.csv", colClasses = c("character"))

  ddr_som_var <- ddr_som_var[, setdiff(colnames(ddr_som_var),
                                       c("COSMIC_total_alterations_in_gene", "selected_KEGG"))]
  colnames(ddr_som_var) <- c("Gene", "Variant Classification", "Protein", "Genomic Change", "VAF", "COS_n_ov",
                             "UniProt_Region", "DNA_repair", "Irino", "TMZ")
  k_tbl <- kable(ddr_som_var, format = "latex", booktabs = TRUE)
  k_tbl <- column_spec(k_tbl, which(colnames(ddr_som_var) == "UniProt_Region"), width = "10em")
  kable_styling(k_tbl, latex_options = "scale_down")
} else {
  cat("No variation to report.")
}
```

Somatic SNVs/indels in important KEGG Pathway genes:

```{r kegg}
if(file.exists("./NOTATES/Somatic_SNV/KEGG_selected.csv")) {
  kegg_som_var <- read.csv("./NOTATES/Somatic_SNV/KEGG_selected.csv", colClasses = c("character"))
  
  colnames(kegg_som_var) <- c("Gene", "Variant Classification", "Protein", "Genomic Change", "VAF", "COS_n_ov",
                              "COS_tot", "UniProt_Region", "DNA_repair", "Irino", "TMZ", "KEGG")
  
  k_tbl <- kable(kegg_som_var, format = "latex", booktabs = TRUE)
  k_tbl <- column_spec(k_tbl, which(colnames(kegg_som_var) == "UniProt_Region"), width = "10em")
  kable_styling(k_tbl, latex_options = "scale_down")
} else {
  cat("No variation to report.")
}
```

\newpage

# IV. Somatic Copy Number Alterations (SCNAs)

Only SCNAs with a $log_2(Tumor/Normal)$ ratio $\le$ -0.25 or $\ge$ 0.25 were used in analysis.

## IV. a. SCNA Burden

Numerous studies have shown that SCNA burden is an important prognostic marker[^7]^,^[^8]^,^[^9]. Below are 4 metrics associated with SCNA Burden:

[^7]: Hieronymus H, Murali R, Tin A, et al. Tumor copy number alteration burden is a pan-cancer prognostic factor associated with recurrence and death. Elife. 2018;7
[^8]: Hieronymus H, Schultz N, Gopalan A, et al. Copy number alteration burden predicts prostate cancer relapse. Proc Natl Acad Sci USA. 2014;111(30):11139-44.s
[^9]: Zhang L, Feizi N, Chi C, Hu P. Association Analysis of Somatic Copy Number Alteration Burden With Breast Cancer Survival. Front Genet. 2018;9:421.

```{r SCNA_burden}
chr_list <- c("chr1", "chr2", "chr3", "chr4", "chr5", "chr6", 
              "chr7", "chr8", "chr9", "chr10", "chr11", "chr12", 
              "chr13", "chr14", "chr15", "chr16", "chr17", "chr18",
              "chr19", "chr20", "chr21", "chr22")

segs_df <- read.csv("NOTATES/SCNA/exomeCNV.csv")
segs_df <- segs_df[segs_df$chr %in% chr_list, ]
segs_df <- segs_df[!is.na(segs_df$logR), ]
segs_df <- segs_df[segs_df$logR > .25 | segs_df$logR < -.25, ]

segs_gr <- makeGRangesFromDataFrame(segs_df, 
                                    seqnames.field = "chr", 
                                    start.field = "probe_start", 
                                    end.field = "probe_end")

exome_df <- read.delim(params$exome_bed, header = FALSE)
exome_df <- exome_df[exome_df$V1 %in% chr_list, ]

exome_gr <- makeGRangesFromDataFrame(exome_df, 
                                     seqnames.field = "V1", 
                                     start.field = "V2", 
                                     end.field = "V3")
instersection_gr <- intersect(segs_gr, exome_gr)

######### Calculate SCNA burden metrics
intersection_df <- as.data.frame(instersection_gr)
### 1. Total altered Length (Mb)
tot_altered_len <- sum(intersection_df$width) / 1e6
### 2. Altered Ratio
altered_ratio <- tot_altered_len / (sum(exome_df$V3 - exome_df$V2 + 1) / 1e6)
### 3. Total Num. Segments
num_segs <- nrow(segs_df)
### 4. Average Altered Length (kb)
ave_alt_len <- tot_altered_len * 1e3 / num_segs
### 5. WGII
ratio_by_chr <- c()
for (chrm in chr_list) {
  seg_chr_df <- intersection_df[intersection_df$seqnames == chrm, ]
  kit_chr_df <- exome_df[exome_df$V1 == chrm, ]
  
  chr_width <- sum(kit_chr_df$V3 - kit_chr_df$V2 + 1)
  ratio_by_chr <- c(ratio_by_chr, 
                    sum(seg_chr_df$width) / chr_width)
}
WGII <- mean(ratio_by_chr)


cnv_burden <- data.frame(tot_altered_len,
                         altered_ratio,
                         num_segs,
                         ave_alt_len,
                         WGII)

######### Report metrics
cnv_burden2 <- cbind(round(tot_altered_len, 2), 
                     round(WGII * 100, 2), 
                     as.character(num_segs), 
                     round(ave_alt_len, 2))
colnames(cnv_burden2) <- c("Total Altered Length (Mbp)", 
                           "Weighted Genome Instability Index (%)", 
                           "Total Number of Alterations", 
                           "Average Length of Alterations (kbp)")
kable_styling(kable(t(cnv_burden2),
                    format = "latex", 
                    booktabs = TRUE, 
                    col.names = NULL,
                    align = "c"))
```

```{r SCNA_burden_save}
SCNA_burden_tbl_path <- file.path(params$script_dir, 
                                  paste0(params$type, 
                                         "_SCNA_burden_table.RDS"))
if (!file.exists(SCNA_burden_tbl_path)) {
  scna_burden_df <- data.frame()
} else {
  scna_burden_df <- readRDS(SCNA_burden_tbl_path)
}

## add to SCNA burden table if not present
if(!params$ID %in% scna_burden_df$ID) {
  current_df <- data.frame(ID = params$ID, 
                           primary = params$primary_tm)
  current_df <- cbind(current_df, cnv_burden)
  scna_burden_df <- rbind(scna_burden_df, current_df)
  saveRDS(scna_burden_df, SCNA_burden_tbl_path)
}
```

`r if(params$type == "glioma") {"## IV. b. Established SCNAs in Glioma"} else {"## IV. b. Established SCNAs in Medulloblastoma"}`

`r if(params$type == "glioma") {"This is the intersection of SCNAs in this case with a list of SCNAs manually curated because of their importance in gliomas."} else {"This is the intersection of SCNAs in this case with a list of SCNAs manually curated because of their importance in medulloblastomas."}`

```{r imp_cna}
if(file.exists("./NOTATES/SCNA/important_CNVs.csv")) {
  imp_cna <- read.csv("./NOTATES/SCNA/important_CNVs.csv")
  kable_styling(kable(imp_cna,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
} else {
  cat("No variation to report.")
}
```

## IV. c. SCNAs in Cancer Gene Census Genes

This subsection lists SCNAs where the gene subject to copy-number alteration is listed in CGC.

```{r cgc_cna}
if(file.exists("./NOTATES/SCNA/CGC_CNVs.csv")) {
  cgc_cna <- read.csv("./NOTATES/SCNA/CGC_CNVs.csv")
  kable_styling(kable(cgc_cna,
                      format = "latex", 
                      booktabs = TRUE,
                      align = "c"))
} else {
  cat("No variation to report.")
}
```

## IV. d. Broad SCNAs

This subsection lists SCNA events that span over one or more cytobands.

```{r brod_cna}
if(file.exists("./NOTATES/SCNA/broad.csv")) {
  broad_cna <- read.csv("./NOTATES/SCNA/broad.csv")
  colnames(broad_cna)[1] <- "chromosome:start-end"
  kable_styling(kable(broad_cna,
                      format = "latex", booktabs = TRUE,
                      align = "c", longtable = TRUE,
                      row.names = FALSE, 
                      linesep = "\\addlinespace"), 
                latex_options = c("hold_position", 
                                  "repeat_header"),
                full_width = TRUE)
} else {
  cat("No variation to report.")
}
```

\newpage

## IV. e. Plots of SCNA Segments by Chromosome

This subsection displays SCNA plots (log2(Tumor/Normal) ratio vs. position) per all chromosomes.

\begin{center}
  \makebox[\textwidth]{\includegraphics[width=\paperwidth]{./ExomeCNV/{CNV.cnv}.png}}
\end{center}

\newpage

# V. Loss of Heterozygosity (LOH) Events

Only LOH events for which absolute difference of B-allele frequencies ($|BAF_{Tumor}-BAF_{Normal}|$) is larger than 0.4 are reported.

## V. a. LOH Overview

All LOH events that pass the filter are reported here.

```{r loh_regions}
loh <- read.csv("./NOTATES/LOH/LOH.csv")
if (nrow(loh) == 0) {
  cat("No variation to report.")
} else {
  loh$Position <- paste0(loh$Chr, ":", loh$Start, "-", loh$End)
  loh <- loh[,c("Genes", "Position", "N_BAF", "T_BAF", "Absolute_Diff")]
  
  kable_styling(kable(loh,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
}
```

## V. b. LOH + Somatic SNV/Indel

Here, alterations where a gene has LOH and a somatic SNV/indel are reported.

```{r loh_dhit}
if(file.exists("./NOTATES/LOH/d_hit_loh.csv"))
{
  loh_dhit <- read.csv("./NOTATES/LOH/d_hit_loh.csv")
  k_tbl <- kable(loh_dhit, format = "latex", booktabs = TRUE, align = "c")
  kable_styling(k_tbl, latex_options = "scale_down")
}else
{
  cat("No event to report.")
}
```


## V. c. LOH Events in Cancer Gene Census Genes 

LOH events where the gene subject to LOH is listed in CGC are reported here.

```{r loh_cgc}
if(file.exists("./NOTATES/LOH/CGC_loh.csv"))
{
  loh_cgc <- read.csv("./NOTATES/LOH/CGC_loh.csv")
  k_tbl <- kable(loh_cgc, format = "latex", booktabs = TRUE, 
                 align = "c", longtable = TRUE,
                 row.names = FALSE, 
                 linesep = "\\addlinespace")
  kable_styling(k_tbl, latex_options = c("hold_position", 
                                         "repeat_header"),
                full_width = TRUE)
} else {
  cat("No event to report.")
}
```

\newpage

# VI. Genes with Double Hit

A double hit strongly suggests a relevant tumor-suppressor gene [^10]. In this section, the list of genes with somatic SNV/indel(s) as well as SCNA and/or LOH events are reported.

[^10]: Vogelstein B, Papadopoulos N, Velculescu VE, Zhou S, Diaz LA, Kinzler KW. Cancer genome landscapes. Science. 2013;339(6127):1546-58.

```{r dhit_tbl}
if(file.exists("./NOTATES/double_hit.csv"))
{
  dhit_tbl <- read.csv("./NOTATES/double_hit.csv")
  k_tbl <- kable(dhit_tbl, format = "latex", booktabs = TRUE, align = "c")
  kable_styling(k_tbl, latex_options = "scale_down")
} else {
  cat("No event to report.")
}
```


# VII. Tumor Heterogeneity Analysis 

```{r n3_or_n2}
if(!file.exists("./THetA/output/CNV.BEST.results")){
   n3_chosen <- F
}else{
  theta_out <- read.delim("./THetA/output/CNV.BEST.results")
  n3_chosen <- ifelse(length(unlist(strsplit(theta_out[1,"mu"], ","))) == 2, F, T)
  
  image_path <- ifelse(n3_chosen,"THetA/output/CNV.n3.graph.pdf","THetA/output/CNV.n2.graph.pdf")
  renamed_path <- sub("\\.g", "_g", sub("\\.n", "_n", image_path))
}
```

THetA (Tumor Heterogeneity Analysis) is an algorithm that estimates the tumor purity and clonal/sublconal copy number aberrations directly from high-throughput DNA sequencing data.

\begin{center} 
\includegraphics[width=500pt]{`r renamed_path`}
\end{center}

\newpage

# VIII. Mutational Signatures

Using DeConstructSigs, the weights of Mutational Signatures v3 (May 2019) from COSMIC are estimated. This section presents the mutational signatures detected in this tumor.

\begin{center} 
\includegraphics[width=500pt]{mut_signatures.pdf}
\end{center}

\begin{center} 
\includegraphics[width=200pt]{mut_signatures_pie.pdf}
\end{center}

```{r mut_sigs}
mut_sigs <- read.csv("./mut_signatures.csv", row.names = 1)
k_tbl <- kable(mut_sigs, format = "latex", booktabs = TRUE)
k_tbl <- column_spec(k_tbl, 3, width = "30em")
k_tbl <- column_spec(k_tbl, 4, width = "30em")
k_tbl <- column_spec(k_tbl, 5, width = "30em")
kable_styling(k_tbl, latex_options = "scale_down")
```

\newpage

# IX. pathfindR - KEGG Pathway Enrichment Analysis

Pathway enrichment analysis is frequently utilized for studying the mechanisms underlying oncological processes. For this analysis KEGG pathway enrichment analysis was performed using the active-subnetwork-oriented enrichment approach of `pathfindR`. `pathfindR` leverages interaction information from a protein-protein interaction network (PIN) to identify distinct active subnetworks ^[An active subnetwork can be defined as a group of interconnected genes in a PIN that predominantly consists of significantly altered genes] and then perform enrichment analyses on these subnetworks. 

## IX. a. Enrichment Results for High-impact Somatic SNV/indels

Genes that harbor non-synonymous mutations with a VAF > 5% and not in FLAGS are included in the enrichment analysis.

\begin{center}
\IfFileExists{pathfindR_results/HQ_mutations/enrichment_chart.png}{\includegraphics[width=300pt]{pathfindR_results/HQ_mutations/enrichment_chart.png}}{}
\end{center}

```{r enr_res_mut, message=FALSE, warning=FALSE}
read_result <- function(f_name) {
  res <- tryCatch(
    {
      read.csv(f_name, row.names = 1)
    }, error = function(cond) {
      message(paste("Cannot read file:", f_name))
      message("Here's the original error message:")
      message(cond)
      return(NA)
    })
  return(res)
}

enr_res_file <- "pathfindR_results/HQ_mutations/mut_enrichment_results.csv"
enr_res <- read_result(enr_res_file)

if (is.data.frame(enr_res)) {
  enr_res$Down_regulated <- NULL
  colnames(enr_res)[7] <- "Somatic SNV/indel"
  kable_styling(kable(enr_res,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
} else {
  cat("There are no enriched KEGG pathways.")
}

```


## IX. b. Enrichment Results for High-impact SCNA

Genes that harbor homozygous deletion or multi-copy amplification (3+) are included in the enrichment analysis.

\begin{center}
\IfFileExists{pathfindR_results/HQ_SCNA/enrichment_chart.png}{\includegraphics[width=300pt]{pathfindR_results/HQ_SCNA/enrichment_chart.png}}{}
\end{center}

```{r enr_res_SCNA, message=FALSE, warning=FALSE}
read_result <- function(f_name) {
  res <- tryCatch(
    {
      read.csv(f_name, row.names = 1)
    }, error = function(cond) {
      message(paste("Cannot read file:", f_name))
      message("Here's the original error message:")
      message(cond)
      return(NA)
    })
  return(res)
}
enr_res_file <- "pathfindR_results/HQ_SCNA/SCNA_enrichment_results.csv"
enr_res <- read_result(enr_res_file)

if (is.data.frame(enr_res)) {
  colnames(enr_res)[7:8] <- c("Amplification", "Deletion")
  kable_styling(kable(enr_res,
                      format = "latex", booktabs = TRUE), 
                latex_options = "scale_down")
} else {
  cat("There are no enriched KEGG pathways.")
}
```
